{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "c77f8da1-2837-4bf7-b3f4-dfcaa7ab42f6",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-11-09T01:15:08.610638Z",
     "iopub.status.busy": "2023-11-09T01:15:08.610520Z",
     "iopub.status.idle": "2023-11-09T01:15:08.692100Z",
     "shell.execute_reply": "2023-11-09T01:15:08.691527Z"
    }
   },
   "outputs": [],
   "source": [
    "import scipy.io\n",
    "import numpy as np\n",
    "import random\n",
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "3fe25376-b856-4296-ba02-5e3771ff9451",
   "metadata": {
    "collapsed": true,
    "execution": {
     "iopub.execute_input": "2023-11-09T01:15:08.694565Z",
     "iopub.status.busy": "2023-11-09T01:15:08.694390Z",
     "iopub.status.idle": "2023-11-09T03:05:56.108350Z",
     "shell.execute_reply": "2023-11-09T03:05:56.107774Z"
    },
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 20:15:08.871855: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.\n",
      "To enable the following instructions: FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.\n",
      "/usr/lib/python3.11/site-packages/h5py/__init__.py:36: UserWarning: h5py is running against HDF5 1.14.2 when it was built against 1.14.1, this may cause problems\n",
      "  _warn((\"h5py is running against HDF5 {0} when it was built against {1}, \"\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "34793\n",
      "WARNING:tensorflow:From /usr/lib/python3.11/site-packages/tensorflow/python/compat/v2_compat.py:107: disable_resource_variables (from tensorflow.python.ops.variable_scope) is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "non-resource variables are not supported in the long term\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 20:16:31.600355: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 20:16:31.611883: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 20:16:31.612128: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 20:16:31.615497: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 20:16:31.615697: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 20:16:31.615867: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 20:16:31.706800: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 20:16:31.707007: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 20:16:31.707177: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 20:16:31.707317: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1639] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 46608 MB memory:  -> device: 0, name: NVIDIA RTX A6000, pci bus id: 0000:2d:00.0, compute capability: 8.6\n",
      "2023-11-08 20:16:31.707623: I tensorflow/core/common_runtime/process_util.cc:146] Creating new thread pool with default inter op setting: 2. Tune using inter_op_parallelism_threads for best performance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 20:16:32.656260: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 20:16:32.656509: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 20:16:32.656683: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 20:16:32.656892: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 20:16:32.657068: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 20:16:32.657210: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1639] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 46608 MB memory:  -> device: 0, name: NVIDIA RTX A6000, pci bus id: 0000:2d:00.0, compute capability: 8.6\n",
      "2023-11-08 20:16:32.657237: I tensorflow/core/common_runtime/process_util.cc:146] Creating new thread pool with default inter op setting: 2. Tune using inter_op_parallelism_threads for best performance.\n",
      "2023-11-08 20:16:32.679099: I tensorflow/compiler/mlir/mlir_graph_optimization_pass.cc:375] MLIR V1 optimization pass is not enabled\n",
      "2023-11-08 20:16:32.896891: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_20/lstm_cell_20/kernel/Assign' id:3316 op device:{requested: '', assigned: ''} def:{{{node lstm_20/lstm_cell_20/kernel/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_20/lstm_cell_20/kernel, lstm_20/lstm_cell_20/kernel/Initializer/random_uniform)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 20:16:33.028848: W tensorflow/c/c_api.cc:304] Operation '{name:'strided_slice/stack_1' id:3634 op device:{requested: '', assigned: ''} def:{{{node strided_slice/stack_1}} = Const[_has_manual_control_dependencies=true, dtype=DT_INT32, value=Tensor<type: int32 shape: [4] values: 0 0 0...>]()}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 20:16:33.062273: W tensorflow/c/c_api.cc:304] Operation '{name:'strided_slice/stack_2' id:3635 op device:{requested: '', assigned: ''} def:{{{node strided_slice/stack_2}} = Const[_has_manual_control_dependencies=true, dtype=DT_INT32, value=Tensor<type: int32 shape: [4] values: 1 1 1...>]()}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "(31305, 95)\n",
      "Train on 31305 samples, validate on 3488 samples\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 20:16:37.608543: W tensorflow/c/c_api.cc:304] Operation '{name:'training/Adam/lstm_34/lstm_cell_34/recurrent_kernel/v/Assign' id:17552 op device:{requested: '', assigned: ''} def:{{{node training/Adam/lstm_34/lstm_cell_34/recurrent_kernel/v/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](training/Adam/lstm_34/lstm_cell_34/recurrent_kernel/v, training/Adam/lstm_34/lstm_cell_34/recurrent_kernel/v/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Epoch 1/50\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 20:16:41.937670: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n",
      "2023-11-08 20:16:44.271121: I tensorflow/compiler/xla/stream_executor/cuda/cuda_blas.cc:606] TensorFloat-32 will be used for the matrix multiplication. This will only be logged once.\n",
      "2023-11-08 20:16:44.283937: I tensorflow/compiler/xla/stream_executor/cuda/cuda_dnn.cc:432] Loaded cuDNN version 8902\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31305/31305 [==============================] - ETA: 0s - loss: 3.2585"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/lib/python3.11/site-packages/keras/engine/training_v1.py:2335: UserWarning: `Model.state_updates` will be removed in a future version. This property should not be used in TensorFlow 2.0, as `updates` are applied automatically.\n",
      "  updates = self.state_updates\n",
      "2023-11-08 20:17:04.274744: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_1/mul' id:6477 op device:{requested: '', assigned: ''} def:{{{node loss_1/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_1/mul/x, loss_1/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Epoch 1: val_loss improved from inf to 1.83078, saving model to ./checkpoints/unknown_person_baseline_p17_20.h5\n",
      "31305/31305 [==============================] - 28s 886us/sample - loss: 3.2585 - val_loss: 1.8308\n",
      "Epoch 2/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.7637\n",
      "Epoch 2: val_loss improved from 1.83078 to 1.60815, saving model to ./checkpoints/unknown_person_baseline_p17_20.h5\n",
      "31305/31305 [==============================] - 21s 670us/sample - loss: 1.7637 - val_loss: 1.6082\n",
      "Epoch 3/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.5828\n",
      "Epoch 3: val_loss improved from 1.60815 to 1.54647, saving model to ./checkpoints/unknown_person_baseline_p17_20.h5\n",
      "31305/31305 [==============================] - 20s 631us/sample - loss: 1.5828 - val_loss: 1.5465\n",
      "Epoch 4/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.5300\n",
      "Epoch 4: val_loss improved from 1.54647 to 1.52184, saving model to ./checkpoints/unknown_person_baseline_p17_20.h5\n",
      "31305/31305 [==============================] - 21s 658us/sample - loss: 1.5300 - val_loss: 1.5218\n",
      "Epoch 5/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.5064\n",
      "Epoch 5: val_loss improved from 1.52184 to 1.50826, saving model to ./checkpoints/unknown_person_baseline_p17_20.h5\n",
      "31305/31305 [==============================] - 22s 701us/sample - loss: 1.5064 - val_loss: 1.5083\n",
      "Epoch 6/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.4908\n",
      "Epoch 6: val_loss improved from 1.50826 to 1.49264, saving model to ./checkpoints/unknown_person_baseline_p17_20.h5\n",
      "31305/31305 [==============================] - 22s 706us/sample - loss: 1.4908 - val_loss: 1.4926\n",
      "Epoch 7/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.4811\n",
      "Epoch 7: val_loss improved from 1.49264 to 1.48504, saving model to ./checkpoints/unknown_person_baseline_p17_20.h5\n",
      "31305/31305 [==============================] - 21s 656us/sample - loss: 1.4811 - val_loss: 1.4850\n",
      "Epoch 8/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.4714\n",
      "Epoch 8: val_loss improved from 1.48504 to 1.47719, saving model to ./checkpoints/unknown_person_baseline_p17_20.h5\n",
      "31305/31305 [==============================] - 23s 719us/sample - loss: 1.4714 - val_loss: 1.4772\n",
      "Epoch 9/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.4718\n",
      "Epoch 9: val_loss improved from 1.47719 to 1.47463, saving model to ./checkpoints/unknown_person_baseline_p17_20.h5\n",
      "31305/31305 [==============================] - 20s 631us/sample - loss: 1.4718 - val_loss: 1.4746\n",
      "Epoch 10/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.4734\n",
      "Epoch 10: val_loss improved from 1.47463 to 1.47044, saving model to ./checkpoints/unknown_person_baseline_p17_20.h5\n",
      "31305/31305 [==============================] - 21s 683us/sample - loss: 1.4734 - val_loss: 1.4704\n",
      "Epoch 11/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.4629\n",
      "Epoch 11: val_loss did not improve from 1.47044\n",
      "31305/31305 [==============================] - 21s 658us/sample - loss: 1.4629 - val_loss: 1.4721\n",
      "Epoch 12/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.4584\n",
      "Epoch 12: val_loss improved from 1.47044 to 1.46273, saving model to ./checkpoints/unknown_person_baseline_p17_20.h5\n",
      "31305/31305 [==============================] - 22s 714us/sample - loss: 1.4584 - val_loss: 1.4627\n",
      "Epoch 13/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.4566\n",
      "Epoch 13: val_loss improved from 1.46273 to 1.45552, saving model to ./checkpoints/unknown_person_baseline_p17_20.h5\n",
      "31305/31305 [==============================] - 22s 695us/sample - loss: 1.4566 - val_loss: 1.4555\n",
      "Epoch 14/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.4648\n",
      "Epoch 14: val_loss did not improve from 1.45552\n",
      "31305/31305 [==============================] - 20s 644us/sample - loss: 1.4648 - val_loss: 1.4719\n",
      "Epoch 15/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.4502\n",
      "Epoch 15: val_loss improved from 1.45552 to 1.45492, saving model to ./checkpoints/unknown_person_baseline_p17_20.h5\n",
      "31305/31305 [==============================] - 23s 721us/sample - loss: 1.4502 - val_loss: 1.4549\n",
      "Epoch 16/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.4479\n",
      "Epoch 16: val_loss improved from 1.45492 to 1.45319, saving model to ./checkpoints/unknown_person_baseline_p17_20.h5\n",
      "31305/31305 [==============================] - 21s 680us/sample - loss: 1.4479 - val_loss: 1.4532\n",
      "Epoch 17/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.4447\n",
      "Epoch 17: val_loss improved from 1.45319 to 1.44884, saving model to ./checkpoints/unknown_person_baseline_p17_20.h5\n",
      "31305/31305 [==============================] - 22s 715us/sample - loss: 1.4447 - val_loss: 1.4488\n",
      "Epoch 18/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.4534\n",
      "Epoch 18: val_loss did not improve from 1.44884\n",
      "31305/31305 [==============================] - 22s 711us/sample - loss: 1.4534 - val_loss: 1.4525\n",
      "Epoch 19/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.4468\n",
      "Epoch 19: val_loss improved from 1.44884 to 1.44803, saving model to ./checkpoints/unknown_person_baseline_p17_20.h5\n",
      "31305/31305 [==============================] - 23s 723us/sample - loss: 1.4468 - val_loss: 1.4480\n",
      "Epoch 20/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.4883\n",
      "Epoch 20: val_loss did not improve from 1.44803\n",
      "31305/31305 [==============================] - 21s 671us/sample - loss: 1.4883 - val_loss: 1.4655\n",
      "Epoch 21/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.4526\n",
      "Epoch 21: val_loss did not improve from 1.44803\n",
      "31305/31305 [==============================] - 23s 740us/sample - loss: 1.4526 - val_loss: 1.4544\n",
      "Epoch 22/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.4495\n",
      "Epoch 22: val_loss did not improve from 1.44803\n",
      "31305/31305 [==============================] - 23s 722us/sample - loss: 1.4495 - val_loss: 1.4485\n",
      "Epoch 23/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.4343\n",
      "Epoch 23: val_loss improved from 1.44803 to 1.44290, saving model to ./checkpoints/unknown_person_baseline_p17_20.h5\n",
      "31305/31305 [==============================] - 21s 663us/sample - loss: 1.4343 - val_loss: 1.4429\n",
      "Epoch 24/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.4264\n",
      "Epoch 24: val_loss improved from 1.44290 to 1.44181, saving model to ./checkpoints/unknown_person_baseline_p17_20.h5\n",
      "31305/31305 [==============================] - 21s 667us/sample - loss: 1.4264 - val_loss: 1.4418\n",
      "Epoch 25/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.4249\n",
      "Epoch 25: val_loss improved from 1.44181 to 1.43672, saving model to ./checkpoints/unknown_person_baseline_p17_20.h5\n",
      "31305/31305 [==============================] - 20s 638us/sample - loss: 1.4249 - val_loss: 1.4367\n",
      "Epoch 26/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.4197\n",
      "Epoch 26: val_loss improved from 1.43672 to 1.43121, saving model to ./checkpoints/unknown_person_baseline_p17_20.h5\n",
      "31305/31305 [==============================] - 19s 623us/sample - loss: 1.4197 - val_loss: 1.4312\n",
      "Epoch 27/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.4174\n",
      "Epoch 27: val_loss did not improve from 1.43121\n",
      "31305/31305 [==============================] - 21s 677us/sample - loss: 1.4174 - val_loss: 1.4326\n",
      "Epoch 28/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.4150\n",
      "Epoch 28: val_loss improved from 1.43121 to 1.42874, saving model to ./checkpoints/unknown_person_baseline_p17_20.h5\n",
      "31305/31305 [==============================] - 22s 715us/sample - loss: 1.4150 - val_loss: 1.4287\n",
      "Epoch 29/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.4113\n",
      "Epoch 29: val_loss did not improve from 1.42874\n",
      "31305/31305 [==============================] - 22s 718us/sample - loss: 1.4113 - val_loss: 1.4291\n",
      "Epoch 30/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.4110\n",
      "Epoch 30: val_loss improved from 1.42874 to 1.41986, saving model to ./checkpoints/unknown_person_baseline_p17_20.h5\n",
      "31305/31305 [==============================] - 22s 717us/sample - loss: 1.4110 - val_loss: 1.4199\n",
      "Epoch 31/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.4070\n",
      "Epoch 31: val_loss improved from 1.41986 to 1.41625, saving model to ./checkpoints/unknown_person_baseline_p17_20.h5\n",
      "31305/31305 [==============================] - 20s 635us/sample - loss: 1.4070 - val_loss: 1.4162\n",
      "Epoch 32/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.4051\n",
      "Epoch 32: val_loss did not improve from 1.41625\n",
      "31305/31305 [==============================] - 20s 625us/sample - loss: 1.4051 - val_loss: 1.4229\n",
      "Epoch 33/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.4022\n",
      "Epoch 33: val_loss improved from 1.41625 to 1.41534, saving model to ./checkpoints/unknown_person_baseline_p17_20.h5\n",
      "31305/31305 [==============================] - 20s 652us/sample - loss: 1.4022 - val_loss: 1.4153\n",
      "Epoch 34/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.4035\n",
      "Epoch 34: val_loss did not improve from 1.41534\n",
      "31305/31305 [==============================] - 20s 623us/sample - loss: 1.4035 - val_loss: 1.4227\n",
      "Epoch 35/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.4014\n",
      "Epoch 35: val_loss improved from 1.41534 to 1.41023, saving model to ./checkpoints/unknown_person_baseline_p17_20.h5\n",
      "31305/31305 [==============================] - 20s 623us/sample - loss: 1.4014 - val_loss: 1.4102\n",
      "Epoch 36/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3990\n",
      "Epoch 36: val_loss improved from 1.41023 to 1.40682, saving model to ./checkpoints/unknown_person_baseline_p17_20.h5\n",
      "31305/31305 [==============================] - 20s 653us/sample - loss: 1.3990 - val_loss: 1.4068\n",
      "Epoch 37/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3937\n",
      "Epoch 37: val_loss did not improve from 1.40682\n",
      "31305/31305 [==============================] - 21s 657us/sample - loss: 1.3937 - val_loss: 1.4146\n",
      "Epoch 38/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3930\n",
      "Epoch 38: val_loss did not improve from 1.40682\n",
      "31305/31305 [==============================] - 20s 653us/sample - loss: 1.3930 - val_loss: 1.4117\n",
      "Epoch 39/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3908\n",
      "Epoch 39: val_loss improved from 1.40682 to 1.40314, saving model to ./checkpoints/unknown_person_baseline_p17_20.h5\n",
      "31305/31305 [==============================] - 20s 646us/sample - loss: 1.3908 - val_loss: 1.4031\n",
      "Epoch 40/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3902\n",
      "Epoch 40: val_loss did not improve from 1.40314\n",
      "31305/31305 [==============================] - 22s 711us/sample - loss: 1.3902 - val_loss: 1.4058\n",
      "Epoch 41/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3869\n",
      "Epoch 41: val_loss did not improve from 1.40314\n",
      "31305/31305 [==============================] - 22s 712us/sample - loss: 1.3869 - val_loss: 1.4059\n",
      "Epoch 42/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3868\n",
      "Epoch 42: val_loss improved from 1.40314 to 1.40014, saving model to ./checkpoints/unknown_person_baseline_p17_20.h5\n",
      "31305/31305 [==============================] - 22s 714us/sample - loss: 1.3868 - val_loss: 1.4001\n",
      "Epoch 43/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3868\n",
      "Epoch 43: val_loss did not improve from 1.40014\n",
      "31305/31305 [==============================] - 22s 692us/sample - loss: 1.3868 - val_loss: 1.4042\n",
      "Epoch 44/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3839\n",
      "Epoch 44: val_loss improved from 1.40014 to 1.39689, saving model to ./checkpoints/unknown_person_baseline_p17_20.h5\n",
      "31305/31305 [==============================] - 22s 704us/sample - loss: 1.3839 - val_loss: 1.3969\n",
      "Epoch 45/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3843\n",
      "Epoch 45: val_loss did not improve from 1.39689\n",
      "31305/31305 [==============================] - 21s 673us/sample - loss: 1.3843 - val_loss: 1.4015\n",
      "Epoch 46/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3823\n",
      "Epoch 46: val_loss did not improve from 1.39689\n",
      "31305/31305 [==============================] - 21s 666us/sample - loss: 1.3823 - val_loss: 1.3971\n",
      "Epoch 47/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3779\n",
      "Epoch 47: val_loss did not improve from 1.39689\n",
      "31305/31305 [==============================] - 22s 707us/sample - loss: 1.3779 - val_loss: 1.3985\n",
      "Epoch 48/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3787\n",
      "Epoch 48: val_loss improved from 1.39689 to 1.39045, saving model to ./checkpoints/unknown_person_baseline_p17_20.h5\n",
      "31305/31305 [==============================] - 21s 666us/sample - loss: 1.3787 - val_loss: 1.3905\n",
      "Epoch 49/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3777\n",
      "Epoch 49: val_loss did not improve from 1.39045\n",
      "31305/31305 [==============================] - 20s 646us/sample - loss: 1.3777 - val_loss: 1.3936\n",
      "Epoch 50/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3763\n",
      "Epoch 50: val_loss did not improve from 1.39045\n",
      "31305/31305 [==============================] - 20s 625us/sample - loss: 1.3763 - val_loss: 1.3937\n",
      "WARNING:tensorflow:From /usr/lib/python3.11/site-packages/tensorflow/python/ops/init_ops.py:94: calling GlorotUniform.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Call initializer instance with the dtype argument instead of passing it to the constructor\n",
      "WARNING:tensorflow:From /usr/lib/python3.11/site-packages/tensorflow/python/ops/init_ops.py:94: calling Zeros.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Call initializer instance with the dtype argument instead of passing it to the constructor\n",
      "WARNING:tensorflow:From /usr/lib/python3.11/site-packages/tensorflow/python/ops/init_ops.py:94: calling Orthogonal.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Call initializer instance with the dtype argument instead of passing it to the constructor\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 20:34:28.530847: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_16_1/lstm_cell_53/recurrent_kernel/Assign' id:21220 op device:{requested: '', assigned: ''} def:{{{node lstm_16_1/lstm_cell_53/recurrent_kernel/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_16_1/lstm_cell_53/recurrent_kernel, lstm_16_1/lstm_cell_53/recurrent_kernel/Initializer/mul_1)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 20:34:30.195781: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_32_1/lstm_cell_69/kernel/m/Assign' id:25732 op device:{requested: '', assigned: ''} def:{{{node lstm_32_1/lstm_cell_69/kernel/m/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_32_1/lstm_cell_69/kernel/m, lstm_32_1/lstm_cell_69/kernel/m/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "/usr/lib/python3.11/site-packages/keras/engine/training_v1.py:2359: UserWarning: `Model.state_updates` will be removed in a future version. This property should not be used in TensorFlow 2.0, as `updates` are applied automatically.\n",
      "  updates=self.state_updates,\n",
      "2023-11-08 20:34:31.509527: W tensorflow/c/c_api.cc:304] Operation '{name:'dropout_4_1/cond/Merge' id:24591 op device:{requested: '', assigned: ''} def:{{{node dropout_4_1/cond/Merge}} = Merge[N=2, T=DT_FLOAT, _has_manual_control_dependencies=true](dropout_4_1/cond/Identity, dropout_4_1/cond/dropout/SelectV2)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1485, 2208)\n",
      "(1514, 2208)\n",
      "(1644, 2208)\n",
      "(1764, 2208)\n",
      "(1836, 2208)\n",
      "(1699, 2208)\n",
      "(1369, 2208)\n",
      "(1766, 2208)\n",
      "(1631, 2208)\n",
      "(1692, 2208)\n",
      "(1550, 2208)\n",
      "(1644, 2208)\n",
      "(1800, 2208)\n",
      "(1872, 2208)\n",
      "(1716, 2208)\n",
      "(1788, 2208)\n",
      "(971, 2208)\n",
      "(1692, 2208)\n",
      "(1872, 2208)\n",
      "{1: 5.848297677427737, 2: 5.444426767847507, 4: 10.0, 5: 4.867147116571411, 6: 4.2663885254138645, 8: 8.495193512335392, 9: 6.835921849997282, 10: 7.248197981421013, 11: 6.889621141917001, 12: 9.193756272429944, 13: 6.564069930385142, 19: 8.36722069540616, 21: 8.836759720692092, 22: 1.0, 25: 7.7674550870815455, 26: 6.280161687832182, 27: 5.00612626658341, 28: 5.5898158938237525, 29: 1.3244064309227317}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2793397/527468659.py:300: DeprecationWarning: Conversion of an array with ndim > 0 to a scalar is deprecated, and will error in future. Ensure you extract a single element from your array before performing this operation. (Deprecated NumPy 1.25.)\n",
      "  w_train[i] = weights_dict[int(p_train[i])]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Train on 31305 samples, validate on 3488 samples\n",
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 20:38:27.631882: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_5/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31305/31305 [==============================] - ETA: 0s - loss: 9.0639\n",
      "Epoch 1: val_loss improved from inf to 1.39727, saving model to ./checkpoints/unknown_person_p17_20.h5\n",
      "31305/31305 [==============================] - 28s 899us/sample - loss: 9.0639 - val_loss: 1.3973\n",
      "Epoch 2/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 9.0502\n",
      "Epoch 2: val_loss did not improve from 1.39727\n",
      "31305/31305 [==============================] - 22s 710us/sample - loss: 9.0502 - val_loss: 1.4121\n",
      "Epoch 3/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.9964\n",
      "Epoch 3: val_loss did not improve from 1.39727\n",
      "31305/31305 [==============================] - 21s 665us/sample - loss: 8.9964 - val_loss: 1.3992\n",
      "Epoch 4/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.9563\n",
      "Epoch 4: val_loss did not improve from 1.39727\n",
      "31305/31305 [==============================] - 22s 699us/sample - loss: 8.9563 - val_loss: 1.4017\n",
      "Epoch 5/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.9301\n",
      "Epoch 5: val_loss improved from 1.39727 to 1.38434, saving model to ./checkpoints/unknown_person_p17_20.h5\n",
      "31305/31305 [==============================] - 22s 718us/sample - loss: 8.9301 - val_loss: 1.3843\n",
      "Epoch 6/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.8802\n",
      "Epoch 6: val_loss did not improve from 1.38434\n",
      "31305/31305 [==============================] - 22s 709us/sample - loss: 8.8802 - val_loss: 1.3926\n",
      "Epoch 7/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.8769\n",
      "Epoch 7: val_loss did not improve from 1.38434\n",
      "31305/31305 [==============================] - 22s 715us/sample - loss: 8.8769 - val_loss: 1.3951\n",
      "Epoch 8/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.8591\n",
      "Epoch 8: val_loss did not improve from 1.38434\n",
      "31305/31305 [==============================] - 22s 702us/sample - loss: 8.8591 - val_loss: 1.3868\n",
      "Epoch 9/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.8452\n",
      "Epoch 9: val_loss did not improve from 1.38434\n",
      "31305/31305 [==============================] - 21s 664us/sample - loss: 8.8452 - val_loss: 1.3972\n",
      "Epoch 10/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.8453\n",
      "Epoch 10: val_loss did not improve from 1.38434\n",
      "31305/31305 [==============================] - 22s 704us/sample - loss: 8.8453 - val_loss: 1.3938\n",
      "Epoch 11/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.8095\n",
      "Epoch 11: val_loss did not improve from 1.38434\n",
      "31305/31305 [==============================] - 22s 703us/sample - loss: 8.8095 - val_loss: 1.3875\n",
      "Epoch 12/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.7943\n",
      "Epoch 12: val_loss did not improve from 1.38434\n",
      "31305/31305 [==============================] - 20s 648us/sample - loss: 8.7943 - val_loss: 1.3939\n",
      "Epoch 13/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.7972\n",
      "Epoch 13: val_loss did not improve from 1.38434\n",
      "31305/31305 [==============================] - 19s 611us/sample - loss: 8.7972 - val_loss: 1.3981\n",
      "Epoch 14/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.8085\n",
      "Epoch 14: val_loss did not improve from 1.38434\n",
      "31305/31305 [==============================] - 19s 610us/sample - loss: 8.8085 - val_loss: 1.3902\n",
      "Epoch 15/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.7883\n",
      "Epoch 15: val_loss improved from 1.38434 to 1.38387, saving model to ./checkpoints/unknown_person_p17_20.h5\n",
      "31305/31305 [==============================] - 22s 694us/sample - loss: 8.7883 - val_loss: 1.3839\n",
      "Epoch 16/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.7838\n",
      "Epoch 16: val_loss did not improve from 1.38387\n",
      "31305/31305 [==============================] - 22s 705us/sample - loss: 8.7838 - val_loss: 1.3873\n",
      "Epoch 17/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.7751\n",
      "Epoch 17: val_loss improved from 1.38387 to 1.38373, saving model to ./checkpoints/unknown_person_p17_20.h5\n",
      "31305/31305 [==============================] - 22s 712us/sample - loss: 8.7751 - val_loss: 1.3837\n",
      "Epoch 18/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.7572\n",
      "Epoch 18: val_loss did not improve from 1.38373\n",
      "31305/31305 [==============================] - 22s 692us/sample - loss: 8.7572 - val_loss: 1.4030\n",
      "Epoch 19/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.7434\n",
      "Epoch 19: val_loss did not improve from 1.38373\n",
      "31305/31305 [==============================] - 19s 614us/sample - loss: 8.7434 - val_loss: 1.3945\n",
      "Epoch 20/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.7260\n",
      "Epoch 20: val_loss did not improve from 1.38373\n",
      "31305/31305 [==============================] - 21s 657us/sample - loss: 8.7260 - val_loss: 1.3994\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 20:45:42.380960: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_17_2/lstm_cell_91/kernel/Assign' id:40758 op device:{requested: '', assigned: ''} def:{{{node lstm_17_2/lstm_cell_91/kernel/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_17_2/lstm_cell_91/kernel, lstm_17_2/lstm_cell_91/kernel/Initializer/random_uniform)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 20:45:44.740953: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_21_2/lstm_cell_95/recurrent_kernel/v/Assign' id:45613 op device:{requested: '', assigned: ''} def:{{{node lstm_21_2/lstm_cell_95/recurrent_kernel/v/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_21_2/lstm_cell_95/recurrent_kernel/v, lstm_21_2/lstm_cell_95/recurrent_kernel/v/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Train on 31305 samples, validate on 3488 samples\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 20:45:49.766313: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_5/mul' id:44579 op device:{requested: '', assigned: ''} def:{{{node loss_5/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_5/mul/x, loss_5/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 20:46:02.225298: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_6/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3755"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 20:46:22.553368: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_5/mul' id:44579 op device:{requested: '', assigned: ''} def:{{{node loss_5/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_5/mul/x, loss_5/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Epoch 1: val_loss improved from inf to 1.39535, saving model to ./checkpoints/unknown_person_baseline_2_p17_20.h5\n",
      "31305/31305 [==============================] - 26s 843us/sample - loss: 1.3755 - val_loss: 1.3954\n",
      "Epoch 2/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3773\n",
      "Epoch 2: val_loss did not improve from 1.39535\n",
      "31305/31305 [==============================] - 19s 621us/sample - loss: 1.3773 - val_loss: 1.4026\n",
      "Epoch 3/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3772\n",
      "Epoch 3: val_loss improved from 1.39535 to 1.38856, saving model to ./checkpoints/unknown_person_baseline_2_p17_20.h5\n",
      "31305/31305 [==============================] - 20s 630us/sample - loss: 1.3772 - val_loss: 1.3886\n",
      "Epoch 4/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3736\n",
      "Epoch 4: val_loss did not improve from 1.38856\n",
      "31305/31305 [==============================] - 20s 646us/sample - loss: 1.3736 - val_loss: 1.3918\n",
      "Epoch 5/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3716\n",
      "Epoch 5: val_loss did not improve from 1.38856\n",
      "31305/31305 [==============================] - 22s 713us/sample - loss: 1.3716 - val_loss: 1.3891\n",
      "Epoch 6/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3699\n",
      "Epoch 6: val_loss improved from 1.38856 to 1.38427, saving model to ./checkpoints/unknown_person_baseline_2_p17_20.h5\n",
      "31305/31305 [==============================] - 22s 718us/sample - loss: 1.3699 - val_loss: 1.3843\n",
      "Epoch 7/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3687\n",
      "Epoch 7: val_loss did not improve from 1.38427\n",
      "31305/31305 [==============================] - 22s 709us/sample - loss: 1.3687 - val_loss: 1.3869\n",
      "Epoch 8/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3658\n",
      "Epoch 8: val_loss did not improve from 1.38427\n",
      "31305/31305 [==============================] - 22s 705us/sample - loss: 1.3658 - val_loss: 1.3934\n",
      "Epoch 9/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3676\n",
      "Epoch 9: val_loss improved from 1.38427 to 1.38185, saving model to ./checkpoints/unknown_person_baseline_2_p17_20.h5\n",
      "31305/31305 [==============================] - 20s 624us/sample - loss: 1.3676 - val_loss: 1.3819\n",
      "Epoch 10/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3654\n",
      "Epoch 10: val_loss did not improve from 1.38185\n",
      "31305/31305 [==============================] - 20s 651us/sample - loss: 1.3654 - val_loss: 1.3907\n",
      "Epoch 11/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3660\n",
      "Epoch 11: val_loss improved from 1.38185 to 1.38138, saving model to ./checkpoints/unknown_person_baseline_2_p17_20.h5\n",
      "31305/31305 [==============================] - 23s 719us/sample - loss: 1.3660 - val_loss: 1.3814\n",
      "Epoch 12/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3615\n",
      "Epoch 12: val_loss did not improve from 1.38138\n",
      "31305/31305 [==============================] - 22s 704us/sample - loss: 1.3615 - val_loss: 1.3887\n",
      "Epoch 13/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3598\n",
      "Epoch 13: val_loss did not improve from 1.38138\n",
      "31305/31305 [==============================] - 22s 707us/sample - loss: 1.3598 - val_loss: 1.3816\n",
      "Epoch 14/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3593\n",
      "Epoch 14: val_loss did not improve from 1.38138\n",
      "31305/31305 [==============================] - 22s 708us/sample - loss: 1.3593 - val_loss: 1.3853\n",
      "Epoch 15/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3598\n",
      "Epoch 15: val_loss did not improve from 1.38138\n",
      "31305/31305 [==============================] - 22s 705us/sample - loss: 1.3598 - val_loss: 1.3867\n",
      "Epoch 16/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3563\n",
      "Epoch 16: val_loss did not improve from 1.38138\n",
      "31305/31305 [==============================] - 19s 610us/sample - loss: 1.3563 - val_loss: 1.3817\n",
      "Epoch 17/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3556\n",
      "Epoch 17: val_loss did not improve from 1.38138\n",
      "31305/31305 [==============================] - 20s 627us/sample - loss: 1.3556 - val_loss: 1.3899\n",
      "Epoch 18/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3577\n",
      "Epoch 18: val_loss did not improve from 1.38138\n",
      "31305/31305 [==============================] - 22s 709us/sample - loss: 1.3577 - val_loss: 1.3877\n",
      "Epoch 19/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3530\n",
      "Epoch 19: val_loss improved from 1.38138 to 1.38134, saving model to ./checkpoints/unknown_person_baseline_2_p17_20.h5\n",
      "31305/31305 [==============================] - 22s 704us/sample - loss: 1.3530 - val_loss: 1.3813\n",
      "Epoch 20/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3527\n",
      "Epoch 20: val_loss improved from 1.38134 to 1.38007, saving model to ./checkpoints/unknown_person_baseline_2_p17_20.h5\n",
      "31305/31305 [==============================] - 22s 712us/sample - loss: 1.3527 - val_loss: 1.3801\n",
      "Train on 31305 samples, validate on 3488 samples\n",
      "Epoch 1/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3508\n",
      "Epoch 1: val_loss improved from inf to 1.37741, saving model to ./checkpoints/unknown_person_baseline_p17_21.h5\n",
      "31305/31305 [==============================] - 22s 712us/sample - loss: 1.3508 - val_loss: 1.3774\n",
      "Epoch 2/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3529\n",
      "Epoch 2: val_loss did not improve from 1.37741\n",
      "31305/31305 [==============================] - 21s 677us/sample - loss: 1.3529 - val_loss: 1.3778\n",
      "Epoch 3/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3486\n",
      "Epoch 3: val_loss did not improve from 1.37741\n",
      "31305/31305 [==============================] - 21s 661us/sample - loss: 1.3486 - val_loss: 1.3779\n",
      "Epoch 4/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3497\n",
      "Epoch 4: val_loss improved from 1.37741 to 1.37700, saving model to ./checkpoints/unknown_person_baseline_p17_21.h5\n",
      "31305/31305 [==============================] - 22s 709us/sample - loss: 1.3497 - val_loss: 1.3770\n",
      "Epoch 5/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3459\n",
      "Epoch 5: val_loss did not improve from 1.37700\n",
      "31305/31305 [==============================] - 21s 656us/sample - loss: 1.3459 - val_loss: 1.3782\n",
      "Epoch 6/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3487\n",
      "Epoch 6: val_loss did not improve from 1.37700\n",
      "31305/31305 [==============================] - 19s 622us/sample - loss: 1.3487 - val_loss: 1.3775\n",
      "Epoch 7/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3428\n",
      "Epoch 7: val_loss improved from 1.37700 to 1.37451, saving model to ./checkpoints/unknown_person_baseline_p17_21.h5\n",
      "31305/31305 [==============================] - 22s 711us/sample - loss: 1.3428 - val_loss: 1.3745\n",
      "Epoch 8/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3483\n",
      "Epoch 8: val_loss did not improve from 1.37451\n",
      "31305/31305 [==============================] - 21s 666us/sample - loss: 1.3483 - val_loss: 1.3746\n",
      "Epoch 9/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3504\n",
      "Epoch 9: val_loss did not improve from 1.37451\n",
      "31305/31305 [==============================] - 22s 704us/sample - loss: 1.3504 - val_loss: 1.3780\n",
      "Epoch 10/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3558\n",
      "Epoch 10: val_loss did not improve from 1.37451\n",
      "31305/31305 [==============================] - 21s 674us/sample - loss: 1.3558 - val_loss: 1.3760\n",
      "Epoch 11/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3659\n",
      "Epoch 11: val_loss did not improve from 1.37451\n",
      "31305/31305 [==============================] - 19s 616us/sample - loss: 1.3659 - val_loss: 1.3801\n",
      "Epoch 12/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3552\n",
      "Epoch 12: val_loss did not improve from 1.37451\n",
      "31305/31305 [==============================] - 19s 614us/sample - loss: 1.3552 - val_loss: 1.3816\n",
      "Epoch 13/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3699\n",
      "Epoch 13: val_loss did not improve from 1.37451\n",
      "31305/31305 [==============================] - 21s 666us/sample - loss: 1.3699 - val_loss: 1.3838\n",
      "Epoch 14/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.4107\n",
      "Epoch 14: val_loss did not improve from 1.37451\n",
      "31305/31305 [==============================] - 21s 683us/sample - loss: 1.4107 - val_loss: 1.3836\n",
      "Epoch 15/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3670\n",
      "Epoch 15: val_loss did not improve from 1.37451\n",
      "31305/31305 [==============================] - 20s 646us/sample - loss: 1.3670 - val_loss: 1.3885\n",
      "Epoch 16/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3669\n",
      "Epoch 16: val_loss did not improve from 1.37451\n",
      "31305/31305 [==============================] - 20s 623us/sample - loss: 1.3669 - val_loss: 1.3883\n",
      "Epoch 17/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3726\n",
      "Epoch 17: val_loss did not improve from 1.37451\n",
      "31305/31305 [==============================] - 19s 620us/sample - loss: 1.3726 - val_loss: 1.3937\n",
      "Epoch 18/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3808\n",
      "Epoch 18: val_loss did not improve from 1.37451\n",
      "31305/31305 [==============================] - 22s 708us/sample - loss: 1.3808 - val_loss: 1.3972\n",
      "Epoch 19/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3983\n",
      "Epoch 19: val_loss did not improve from 1.37451\n",
      "31305/31305 [==============================] - 21s 672us/sample - loss: 1.3983 - val_loss: 1.3936\n",
      "Epoch 20/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.4007\n",
      "Epoch 20: val_loss did not improve from 1.37451\n",
      "31305/31305 [==============================] - 19s 620us/sample - loss: 1.4007 - val_loss: 1.3962\n",
      "Epoch 21/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.4152\n",
      "Epoch 21: val_loss did not improve from 1.37451\n",
      "31305/31305 [==============================] - 22s 710us/sample - loss: 1.4152 - val_loss: 1.4111\n",
      "Epoch 22/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3953\n",
      "Epoch 22: val_loss did not improve from 1.37451\n",
      "31305/31305 [==============================] - 22s 712us/sample - loss: 1.3953 - val_loss: 1.3987\n",
      "Epoch 23/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3975\n",
      "Epoch 23: val_loss did not improve from 1.37451\n",
      "31305/31305 [==============================] - 22s 713us/sample - loss: 1.3975 - val_loss: 1.4085\n",
      "Epoch 24/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3846\n",
      "Epoch 24: val_loss did not improve from 1.37451\n",
      "31305/31305 [==============================] - 21s 657us/sample - loss: 1.3846 - val_loss: 1.4053\n",
      "Epoch 25/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3808\n",
      "Epoch 25: val_loss did not improve from 1.37451\n",
      "31305/31305 [==============================] - 21s 667us/sample - loss: 1.3808 - val_loss: 1.3948\n",
      "Epoch 26/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3763\n",
      "Epoch 26: val_loss did not improve from 1.37451\n",
      "31305/31305 [==============================] - 21s 656us/sample - loss: 1.3763 - val_loss: 1.3985\n",
      "Epoch 27/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3751\n",
      "Epoch 27: val_loss did not improve from 1.37451\n",
      "31305/31305 [==============================] - 22s 713us/sample - loss: 1.3751 - val_loss: 1.3977\n",
      "Epoch 28/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3748\n",
      "Epoch 28: val_loss did not improve from 1.37451\n",
      "31305/31305 [==============================] - 22s 718us/sample - loss: 1.3748 - val_loss: 1.3978\n",
      "Epoch 29/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3695\n",
      "Epoch 29: val_loss did not improve from 1.37451\n",
      "31305/31305 [==============================] - 22s 708us/sample - loss: 1.3695 - val_loss: 1.3955\n",
      "Epoch 30/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3689\n",
      "Epoch 30: val_loss did not improve from 1.37451\n",
      "31305/31305 [==============================] - 22s 706us/sample - loss: 1.3689 - val_loss: 1.3922\n",
      "Epoch 31/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3667\n",
      "Epoch 31: val_loss did not improve from 1.37451\n",
      "31305/31305 [==============================] - 21s 678us/sample - loss: 1.3667 - val_loss: 1.3935\n",
      "Epoch 32/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3655\n",
      "Epoch 32: val_loss did not improve from 1.37451\n",
      "31305/31305 [==============================] - 22s 714us/sample - loss: 1.3655 - val_loss: 1.3931\n",
      "Epoch 33/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3604\n",
      "Epoch 33: val_loss did not improve from 1.37451\n",
      "31305/31305 [==============================] - 22s 714us/sample - loss: 1.3604 - val_loss: 1.3979\n",
      "Epoch 34/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3634\n",
      "Epoch 34: val_loss did not improve from 1.37451\n",
      "31305/31305 [==============================] - 22s 712us/sample - loss: 1.3634 - val_loss: 1.3914\n",
      "Epoch 35/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3632\n",
      "Epoch 35: val_loss did not improve from 1.37451\n",
      "31305/31305 [==============================] - 22s 709us/sample - loss: 1.3632 - val_loss: 1.3942\n",
      "Epoch 36/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3579\n",
      "Epoch 36: val_loss did not improve from 1.37451\n",
      "31305/31305 [==============================] - 20s 652us/sample - loss: 1.3579 - val_loss: 1.3895\n",
      "Epoch 37/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3577\n",
      "Epoch 37: val_loss did not improve from 1.37451\n",
      "31305/31305 [==============================] - 19s 619us/sample - loss: 1.3577 - val_loss: 1.3919\n",
      "Epoch 38/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3568\n",
      "Epoch 38: val_loss did not improve from 1.37451\n",
      "31305/31305 [==============================] - 20s 623us/sample - loss: 1.3568 - val_loss: 1.3914\n",
      "Epoch 39/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3535\n",
      "Epoch 39: val_loss did not improve from 1.37451\n",
      "31305/31305 [==============================] - 19s 614us/sample - loss: 1.3535 - val_loss: 1.3943\n",
      "Epoch 40/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3524\n",
      "Epoch 40: val_loss did not improve from 1.37451\n",
      "31305/31305 [==============================] - 19s 619us/sample - loss: 1.3524 - val_loss: 1.3923\n",
      "Epoch 41/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3521\n",
      "Epoch 41: val_loss did not improve from 1.37451\n",
      "31305/31305 [==============================] - 19s 620us/sample - loss: 1.3521 - val_loss: 1.3883\n",
      "Epoch 42/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3522\n",
      "Epoch 42: val_loss did not improve from 1.37451\n",
      "31305/31305 [==============================] - 19s 614us/sample - loss: 1.3522 - val_loss: 1.3871\n",
      "Epoch 43/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3490\n",
      "Epoch 43: val_loss did not improve from 1.37451\n",
      "31305/31305 [==============================] - 20s 644us/sample - loss: 1.3490 - val_loss: 1.3940\n",
      "Epoch 44/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3491\n",
      "Epoch 44: val_loss did not improve from 1.37451\n",
      "31305/31305 [==============================] - 21s 675us/sample - loss: 1.3491 - val_loss: 1.3944\n",
      "Epoch 45/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3482\n",
      "Epoch 45: val_loss did not improve from 1.37451\n",
      "31305/31305 [==============================] - 20s 633us/sample - loss: 1.3482 - val_loss: 1.3905\n",
      "Epoch 46/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3474\n",
      "Epoch 46: val_loss did not improve from 1.37451\n",
      "31305/31305 [==============================] - 20s 635us/sample - loss: 1.3474 - val_loss: 1.3855\n",
      "Epoch 47/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3469\n",
      "Epoch 47: val_loss did not improve from 1.37451\n",
      "31305/31305 [==============================] - 19s 619us/sample - loss: 1.3469 - val_loss: 1.3921\n",
      "Epoch 48/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3474\n",
      "Epoch 48: val_loss did not improve from 1.37451\n",
      "31305/31305 [==============================] - 21s 681us/sample - loss: 1.3474 - val_loss: 1.3961\n",
      "Epoch 49/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3477\n",
      "Epoch 49: val_loss did not improve from 1.37451\n",
      "31305/31305 [==============================] - 21s 662us/sample - loss: 1.3477 - val_loss: 1.3906\n",
      "Epoch 50/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3432\n",
      "Epoch 50: val_loss did not improve from 1.37451\n",
      "31305/31305 [==============================] - 19s 610us/sample - loss: 1.3432 - val_loss: 1.3852\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 21:10:36.802544: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_22_3/lstm_cell_133/recurrent_kernel/Assign' id:60940 op device:{requested: '', assigned: ''} def:{{{node lstm_22_3/lstm_cell_133/recurrent_kernel/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_22_3/lstm_cell_133/recurrent_kernel, lstm_22_3/lstm_cell_133/recurrent_kernel/Initializer/mul_1)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 21:10:40.077748: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_31_3/lstm_cell_142/kernel/v/Assign' id:65118 op device:{requested: '', assigned: ''} def:{{{node lstm_31_3/lstm_cell_142/kernel/v/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_31_3/lstm_cell_142/kernel/v, lstm_31_3/lstm_cell_142/kernel/v/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 21:10:43.230263: W tensorflow/c/c_api.cc:304] Operation '{name:'dropout_4_3/cond/Merge' id:63349 op device:{requested: '', assigned: ''} def:{{{node dropout_4_3/cond/Merge}} = Merge[N=2, T=DT_FLOAT, _has_manual_control_dependencies=true](dropout_4_3/cond/Identity, dropout_4_3/cond/dropout/SelectV2)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1485, 2208)\n",
      "(1514, 2208)\n",
      "(1644, 2208)\n",
      "(1764, 2208)\n",
      "(1836, 2208)\n",
      "(1699, 2208)\n",
      "(1369, 2208)\n",
      "(1766, 2208)\n",
      "(1631, 2208)\n",
      "(1692, 2208)\n",
      "(1550, 2208)\n",
      "(1644, 2208)\n",
      "(1800, 2208)\n",
      "(1872, 2208)\n",
      "(1716, 2208)\n",
      "(1788, 2208)\n",
      "(971, 2208)\n",
      "(1692, 2208)\n",
      "(1872, 2208)\n",
      "{1: 5.841983499934524, 2: 5.226435173505195, 4: 10.0, 5: 4.915454883766353, 6: 4.0940512610320425, 8: 8.611975605768142, 9: 6.559507291561085, 10: 7.299228265702696, 11: 6.64502530646308, 12: 9.339860213327453, 13: 6.305074683844925, 19: 8.443322030938917, 21: 9.040298128705246, 22: 1.0649435385841255, 25: 7.8251262363766445, 26: 6.404846145138725, 27: 4.456562626929983, 28: 5.619660099539002, 29: 1.0}\n",
      "Train on 31305 samples, validate on 3488 samples\n",
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 21:14:41.358321: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_7/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31305/31305 [==============================] - ETA: 0s - loss: 8.9089\n",
      "Epoch 1: val_loss improved from inf to 1.39965, saving model to ./checkpoints/unknown_person_p17_21.h5\n",
      "31305/31305 [==============================] - 31s 997us/sample - loss: 8.9089 - val_loss: 1.3997\n",
      "Epoch 2/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.8028\n",
      "Epoch 2: val_loss improved from 1.39965 to 1.38326, saving model to ./checkpoints/unknown_person_p17_21.h5\n",
      "31305/31305 [==============================] - 22s 707us/sample - loss: 8.8028 - val_loss: 1.3833\n",
      "Epoch 3/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.8424\n",
      "Epoch 3: val_loss did not improve from 1.38326\n",
      "31305/31305 [==============================] - 22s 707us/sample - loss: 8.8424 - val_loss: 1.3860\n",
      "Epoch 4/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.7605\n",
      "Epoch 4: val_loss did not improve from 1.38326\n",
      "31305/31305 [==============================] - 22s 706us/sample - loss: 8.7605 - val_loss: 1.3976\n",
      "Epoch 5/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.7338\n",
      "Epoch 5: val_loss did not improve from 1.38326\n",
      "31305/31305 [==============================] - 22s 708us/sample - loss: 8.7338 - val_loss: 1.3833\n",
      "Epoch 6/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.7370\n",
      "Epoch 6: val_loss did not improve from 1.38326\n",
      "31305/31305 [==============================] - 21s 681us/sample - loss: 8.7370 - val_loss: 1.3926\n",
      "Epoch 7/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.6975\n",
      "Epoch 7: val_loss did not improve from 1.38326\n",
      "31305/31305 [==============================] - 19s 615us/sample - loss: 8.6975 - val_loss: 1.3947\n",
      "Epoch 8/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.6742\n",
      "Epoch 8: val_loss did not improve from 1.38326\n",
      "31305/31305 [==============================] - 19s 610us/sample - loss: 8.6742 - val_loss: 1.3837\n",
      "Epoch 9/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.6903\n",
      "Epoch 9: val_loss improved from 1.38326 to 1.38227, saving model to ./checkpoints/unknown_person_p17_21.h5\n",
      "31305/31305 [==============================] - 20s 641us/sample - loss: 8.6903 - val_loss: 1.3823\n",
      "Epoch 10/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.6786\n",
      "Epoch 10: val_loss improved from 1.38227 to 1.37844, saving model to ./checkpoints/unknown_person_p17_21.h5\n",
      "31305/31305 [==============================] - 22s 690us/sample - loss: 8.6786 - val_loss: 1.3784\n",
      "Epoch 11/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.6500\n",
      "Epoch 11: val_loss did not improve from 1.37844\n",
      "31305/31305 [==============================] - 22s 699us/sample - loss: 8.6500 - val_loss: 1.3788\n",
      "Epoch 12/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.6429\n",
      "Epoch 12: val_loss did not improve from 1.37844\n",
      "31305/31305 [==============================] - 20s 651us/sample - loss: 8.6429 - val_loss: 1.3831\n",
      "Epoch 13/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.6402\n",
      "Epoch 13: val_loss did not improve from 1.37844\n",
      "31305/31305 [==============================] - 20s 636us/sample - loss: 8.6402 - val_loss: 1.3842\n",
      "Epoch 14/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.6347\n",
      "Epoch 14: val_loss did not improve from 1.37844\n",
      "31305/31305 [==============================] - 20s 623us/sample - loss: 8.6347 - val_loss: 1.3864\n",
      "Epoch 15/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.6212\n",
      "Epoch 15: val_loss did not improve from 1.37844\n",
      "31305/31305 [==============================] - 19s 620us/sample - loss: 8.6212 - val_loss: 1.3802\n",
      "Epoch 16/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.6046\n",
      "Epoch 16: val_loss did not improve from 1.37844\n",
      "31305/31305 [==============================] - 20s 639us/sample - loss: 8.6046 - val_loss: 1.3817\n",
      "Epoch 17/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.6054\n",
      "Epoch 17: val_loss improved from 1.37844 to 1.37745, saving model to ./checkpoints/unknown_person_p17_21.h5\n",
      "31305/31305 [==============================] - 19s 616us/sample - loss: 8.6054 - val_loss: 1.3775\n",
      "Epoch 18/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.5908\n",
      "Epoch 18: val_loss improved from 1.37745 to 1.37634, saving model to ./checkpoints/unknown_person_p17_21.h5\n",
      "31305/31305 [==============================] - 20s 628us/sample - loss: 8.5908 - val_loss: 1.3763\n",
      "Epoch 19/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.5858\n",
      "Epoch 19: val_loss did not improve from 1.37634\n",
      "31305/31305 [==============================] - 19s 601us/sample - loss: 8.5858 - val_loss: 1.3844\n",
      "Epoch 20/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.5953\n",
      "Epoch 20: val_loss did not improve from 1.37634\n",
      "31305/31305 [==============================] - 21s 679us/sample - loss: 8.5953 - val_loss: 1.3848\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 21:21:44.633398: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_33_4/lstm_cell_181/recurrent_kernel/Assign' id:82098 op device:{requested: '', assigned: ''} def:{{{node lstm_33_4/lstm_cell_181/recurrent_kernel/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_33_4/lstm_cell_181/recurrent_kernel, lstm_33_4/lstm_cell_181/recurrent_kernel/Initializer/mul_1)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 21:21:48.709616: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_29_4/lstm_cell_177/kernel/m/Assign' id:83843 op device:{requested: '', assigned: ''} def:{{{node lstm_29_4/lstm_cell_177/kernel/m/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_29_4/lstm_cell_177/kernel/m, lstm_29_4/lstm_cell_177/kernel/m/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Train on 31305 samples, validate on 3488 samples\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 21:21:55.174782: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_9/mul' id:83337 op device:{requested: '', assigned: ''} def:{{{node loss_9/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_9/mul/x, loss_9/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 21:22:16.579092: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_8/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3540"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 21:22:40.175399: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_9/mul' id:83337 op device:{requested: '', assigned: ''} def:{{{node loss_9/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_9/mul/x, loss_9/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Epoch 1: val_loss improved from inf to 1.37737, saving model to ./checkpoints/unknown_person_baseline_2_p17_21.h5\n",
      "31305/31305 [==============================] - 33s 1ms/sample - loss: 1.3540 - val_loss: 1.3774\n",
      "Epoch 2/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3420\n",
      "Epoch 2: val_loss did not improve from 1.37737\n",
      "31305/31305 [==============================] - 22s 695us/sample - loss: 1.3420 - val_loss: 1.3798\n",
      "Epoch 3/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3549\n",
      "Epoch 3: val_loss did not improve from 1.37737\n",
      "31305/31305 [==============================] - 22s 711us/sample - loss: 1.3549 - val_loss: 1.3824\n",
      "Epoch 4/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3494\n",
      "Epoch 4: val_loss did not improve from 1.37737\n",
      "31305/31305 [==============================] - 22s 709us/sample - loss: 1.3494 - val_loss: 1.3802\n",
      "Epoch 5/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3452\n",
      "Epoch 5: val_loss improved from 1.37737 to 1.37416, saving model to ./checkpoints/unknown_person_baseline_2_p17_21.h5\n",
      "31305/31305 [==============================] - 22s 712us/sample - loss: 1.3452 - val_loss: 1.3742\n",
      "Epoch 6/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3405\n",
      "Epoch 6: val_loss did not improve from 1.37416\n",
      "31305/31305 [==============================] - 22s 709us/sample - loss: 1.3405 - val_loss: 1.3784\n",
      "Epoch 7/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3406\n",
      "Epoch 7: val_loss did not improve from 1.37416\n",
      "31305/31305 [==============================] - 21s 670us/sample - loss: 1.3406 - val_loss: 1.3771\n",
      "Epoch 8/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3413\n",
      "Epoch 8: val_loss improved from 1.37416 to 1.37385, saving model to ./checkpoints/unknown_person_baseline_2_p17_21.h5\n",
      "31305/31305 [==============================] - 20s 624us/sample - loss: 1.3413 - val_loss: 1.3739\n",
      "Epoch 9/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3390\n",
      "Epoch 9: val_loss did not improve from 1.37385\n",
      "31305/31305 [==============================] - 19s 616us/sample - loss: 1.3390 - val_loss: 1.3753\n",
      "Epoch 10/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3379\n",
      "Epoch 10: val_loss did not improve from 1.37385\n",
      "31305/31305 [==============================] - 19s 617us/sample - loss: 1.3379 - val_loss: 1.3853\n",
      "Epoch 11/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3384\n",
      "Epoch 11: val_loss did not improve from 1.37385\n",
      "31305/31305 [==============================] - 19s 619us/sample - loss: 1.3384 - val_loss: 1.3770\n",
      "Epoch 12/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3337\n",
      "Epoch 12: val_loss did not improve from 1.37385\n",
      "31305/31305 [==============================] - 19s 616us/sample - loss: 1.3337 - val_loss: 1.3763\n",
      "Epoch 13/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3375\n",
      "Epoch 13: val_loss did not improve from 1.37385\n",
      "31305/31305 [==============================] - 19s 618us/sample - loss: 1.3375 - val_loss: 1.3770\n",
      "Epoch 14/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3360\n",
      "Epoch 14: val_loss did not improve from 1.37385\n",
      "31305/31305 [==============================] - 22s 709us/sample - loss: 1.3360 - val_loss: 1.3805\n",
      "Epoch 15/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3352\n",
      "Epoch 15: val_loss did not improve from 1.37385\n",
      "31305/31305 [==============================] - 21s 676us/sample - loss: 1.3352 - val_loss: 1.3757\n",
      "Epoch 16/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3322\n",
      "Epoch 16: val_loss improved from 1.37385 to 1.37286, saving model to ./checkpoints/unknown_person_baseline_2_p17_21.h5\n",
      "31305/31305 [==============================] - 20s 627us/sample - loss: 1.3322 - val_loss: 1.3729\n",
      "Epoch 17/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3330\n",
      "Epoch 17: val_loss did not improve from 1.37286\n",
      "31305/31305 [==============================] - 19s 614us/sample - loss: 1.3330 - val_loss: 1.3789\n",
      "Epoch 18/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3335\n",
      "Epoch 18: val_loss did not improve from 1.37286\n",
      "31305/31305 [==============================] - 20s 644us/sample - loss: 1.3335 - val_loss: 1.3779\n",
      "Epoch 19/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3330\n",
      "Epoch 19: val_loss did not improve from 1.37286\n",
      "31305/31305 [==============================] - 19s 622us/sample - loss: 1.3330 - val_loss: 1.3748\n",
      "Epoch 20/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3331\n",
      "Epoch 20: val_loss improved from 1.37286 to 1.36895, saving model to ./checkpoints/unknown_person_baseline_2_p17_21.h5\n",
      "31305/31305 [==============================] - 19s 618us/sample - loss: 1.3331 - val_loss: 1.3690\n",
      "Train on 31305 samples, validate on 3488 samples\n",
      "Epoch 1/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3309\n",
      "Epoch 1: val_loss improved from inf to 1.37061, saving model to ./checkpoints/unknown_person_baseline_p17_22.h5\n",
      "31305/31305 [==============================] - 19s 613us/sample - loss: 1.3309 - val_loss: 1.3706\n",
      "Epoch 2/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3300\n",
      "Epoch 2: val_loss did not improve from 1.37061\n",
      "31305/31305 [==============================] - 19s 619us/sample - loss: 1.3300 - val_loss: 1.3744\n",
      "Epoch 3/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3289\n",
      "Epoch 3: val_loss improved from 1.37061 to 1.36920, saving model to ./checkpoints/unknown_person_baseline_p17_22.h5\n",
      "31305/31305 [==============================] - 20s 638us/sample - loss: 1.3289 - val_loss: 1.3692\n",
      "Epoch 4/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3259\n",
      "Epoch 4: val_loss did not improve from 1.36920\n",
      "31305/31305 [==============================] - 19s 611us/sample - loss: 1.3259 - val_loss: 1.3756\n",
      "Epoch 5/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3286\n",
      "Epoch 5: val_loss did not improve from 1.36920\n",
      "31305/31305 [==============================] - 19s 610us/sample - loss: 1.3286 - val_loss: 1.3749\n",
      "Epoch 6/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3254\n",
      "Epoch 6: val_loss improved from 1.36920 to 1.36816, saving model to ./checkpoints/unknown_person_baseline_p17_22.h5\n",
      "31305/31305 [==============================] - 20s 625us/sample - loss: 1.3254 - val_loss: 1.3682\n",
      "Epoch 7/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3235\n",
      "Epoch 7: val_loss did not improve from 1.36816\n",
      "31305/31305 [==============================] - 19s 616us/sample - loss: 1.3235 - val_loss: 1.3700\n",
      "Epoch 8/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3353\n",
      "Epoch 8: val_loss did not improve from 1.36816\n",
      "31305/31305 [==============================] - 19s 620us/sample - loss: 1.3353 - val_loss: 1.3797\n",
      "Epoch 9/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3404\n",
      "Epoch 9: val_loss did not improve from 1.36816\n",
      "31305/31305 [==============================] - 22s 702us/sample - loss: 1.3404 - val_loss: 1.3771\n",
      "Epoch 10/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3925\n",
      "Epoch 10: val_loss did not improve from 1.36816\n",
      "31305/31305 [==============================] - 21s 681us/sample - loss: 1.3925 - val_loss: 1.3888\n",
      "Epoch 11/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3554\n",
      "Epoch 11: val_loss did not improve from 1.36816\n",
      "31305/31305 [==============================] - 19s 616us/sample - loss: 1.3554 - val_loss: 1.3806\n",
      "Epoch 12/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.4417\n",
      "Epoch 12: val_loss did not improve from 1.36816\n",
      "31305/31305 [==============================] - 19s 616us/sample - loss: 1.4417 - val_loss: 1.4033\n",
      "Epoch 13/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3616\n",
      "Epoch 13: val_loss did not improve from 1.36816\n",
      "31305/31305 [==============================] - 20s 641us/sample - loss: 1.3616 - val_loss: 1.3839\n",
      "Epoch 14/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.4002\n",
      "Epoch 14: val_loss did not improve from 1.36816\n",
      "31305/31305 [==============================] - 21s 681us/sample - loss: 1.4002 - val_loss: 1.3929\n",
      "Epoch 15/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3873\n",
      "Epoch 15: val_loss did not improve from 1.36816\n",
      "31305/31305 [==============================] - 19s 617us/sample - loss: 1.3873 - val_loss: 1.3955\n",
      "Epoch 16/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3804\n",
      "Epoch 16: val_loss did not improve from 1.36816\n",
      "31305/31305 [==============================] - 19s 615us/sample - loss: 1.3804 - val_loss: 1.4009\n",
      "Epoch 17/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3771\n",
      "Epoch 17: val_loss did not improve from 1.36816\n",
      "31305/31305 [==============================] - 19s 613us/sample - loss: 1.3771 - val_loss: 1.3964\n",
      "Epoch 18/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.4377\n",
      "Epoch 18: val_loss did not improve from 1.36816\n",
      "31305/31305 [==============================] - 19s 614us/sample - loss: 1.4377 - val_loss: 1.4090\n",
      "Epoch 19/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.4753\n",
      "Epoch 19: val_loss did not improve from 1.36816\n",
      "31305/31305 [==============================] - 20s 653us/sample - loss: 1.4753 - val_loss: 1.4102\n",
      "Epoch 20/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.4355\n",
      "Epoch 20: val_loss did not improve from 1.36816\n",
      "31305/31305 [==============================] - 21s 663us/sample - loss: 1.4355 - val_loss: 1.4070\n",
      "Epoch 21/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.4144\n",
      "Epoch 21: val_loss did not improve from 1.36816\n",
      "31305/31305 [==============================] - 19s 610us/sample - loss: 1.4144 - val_loss: 1.4099\n",
      "Epoch 22/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.4157\n",
      "Epoch 22: val_loss did not improve from 1.36816\n",
      "31305/31305 [==============================] - 21s 660us/sample - loss: 1.4157 - val_loss: 1.4144\n",
      "Epoch 23/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.4061\n",
      "Epoch 23: val_loss did not improve from 1.36816\n",
      "31305/31305 [==============================] - 19s 615us/sample - loss: 1.4061 - val_loss: 1.4149\n",
      "Epoch 24/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.4036\n",
      "Epoch 24: val_loss did not improve from 1.36816\n",
      "31305/31305 [==============================] - 20s 655us/sample - loss: 1.4036 - val_loss: 1.4153\n",
      "Epoch 25/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.4056\n",
      "Epoch 25: val_loss did not improve from 1.36816\n",
      "31305/31305 [==============================] - 22s 700us/sample - loss: 1.4056 - val_loss: 1.4087\n",
      "Epoch 26/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3970\n",
      "Epoch 26: val_loss did not improve from 1.36816\n",
      "31305/31305 [==============================] - 22s 702us/sample - loss: 1.3970 - val_loss: 1.4114\n",
      "Epoch 27/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3876\n",
      "Epoch 27: val_loss did not improve from 1.36816\n",
      "31305/31305 [==============================] - 22s 699us/sample - loss: 1.3876 - val_loss: 1.4111\n",
      "Epoch 28/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3856\n",
      "Epoch 28: val_loss did not improve from 1.36816\n",
      "31305/31305 [==============================] - 22s 704us/sample - loss: 1.3856 - val_loss: 1.4099\n",
      "Epoch 29/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3803\n",
      "Epoch 29: val_loss did not improve from 1.36816\n",
      "31305/31305 [==============================] - 21s 686us/sample - loss: 1.3803 - val_loss: 1.4069\n",
      "Epoch 30/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3784\n",
      "Epoch 30: val_loss did not improve from 1.36816\n",
      "31305/31305 [==============================] - 19s 615us/sample - loss: 1.3784 - val_loss: 1.4090\n",
      "Epoch 31/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3740\n",
      "Epoch 31: val_loss did not improve from 1.36816\n",
      "31305/31305 [==============================] - 21s 659us/sample - loss: 1.3740 - val_loss: 1.4082\n",
      "Epoch 32/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3731\n",
      "Epoch 32: val_loss did not improve from 1.36816\n",
      "31305/31305 [==============================] - 22s 702us/sample - loss: 1.3731 - val_loss: 1.4056\n",
      "Epoch 33/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3779\n",
      "Epoch 33: val_loss did not improve from 1.36816\n",
      "31305/31305 [==============================] - 21s 675us/sample - loss: 1.3779 - val_loss: 1.4004\n",
      "Epoch 34/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3739\n",
      "Epoch 34: val_loss did not improve from 1.36816\n",
      "31305/31305 [==============================] - 20s 646us/sample - loss: 1.3739 - val_loss: 1.4026\n",
      "Epoch 35/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3693\n",
      "Epoch 35: val_loss did not improve from 1.36816\n",
      "31305/31305 [==============================] - 22s 701us/sample - loss: 1.3693 - val_loss: 1.4019\n",
      "Epoch 36/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3676\n",
      "Epoch 36: val_loss did not improve from 1.36816\n",
      "31305/31305 [==============================] - 22s 703us/sample - loss: 1.3676 - val_loss: 1.3980\n",
      "Epoch 37/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3660\n",
      "Epoch 37: val_loss did not improve from 1.36816\n",
      "31305/31305 [==============================] - 21s 659us/sample - loss: 1.3660 - val_loss: 1.4009\n",
      "Epoch 38/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3694\n",
      "Epoch 38: val_loss did not improve from 1.36816\n",
      "31305/31305 [==============================] - 20s 649us/sample - loss: 1.3694 - val_loss: 1.3995\n",
      "Epoch 39/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3633\n",
      "Epoch 39: val_loss did not improve from 1.36816\n",
      "31305/31305 [==============================] - 22s 696us/sample - loss: 1.3633 - val_loss: 1.4008\n",
      "Epoch 40/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3634\n",
      "Epoch 40: val_loss did not improve from 1.36816\n",
      "31305/31305 [==============================] - 22s 702us/sample - loss: 1.3634 - val_loss: 1.3956\n",
      "Epoch 41/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3653\n",
      "Epoch 41: val_loss did not improve from 1.36816\n",
      "31305/31305 [==============================] - 21s 668us/sample - loss: 1.3653 - val_loss: 1.3980\n",
      "Epoch 42/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3621\n",
      "Epoch 42: val_loss did not improve from 1.36816\n",
      "31305/31305 [==============================] - 22s 704us/sample - loss: 1.3621 - val_loss: 1.3991\n",
      "Epoch 43/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3620\n",
      "Epoch 43: val_loss did not improve from 1.36816\n",
      "31305/31305 [==============================] - 22s 698us/sample - loss: 1.3620 - val_loss: 1.3986\n",
      "Epoch 44/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3613\n",
      "Epoch 44: val_loss did not improve from 1.36816\n",
      "31305/31305 [==============================] - 20s 643us/sample - loss: 1.3613 - val_loss: 1.4011\n",
      "Epoch 45/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3596\n",
      "Epoch 45: val_loss did not improve from 1.36816\n",
      "31305/31305 [==============================] - 22s 709us/sample - loss: 1.3596 - val_loss: 1.3983\n",
      "Epoch 46/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3564\n",
      "Epoch 46: val_loss did not improve from 1.36816\n",
      "31305/31305 [==============================] - 22s 708us/sample - loss: 1.3564 - val_loss: 1.4045\n",
      "Epoch 47/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3600\n",
      "Epoch 47: val_loss did not improve from 1.36816\n",
      "31305/31305 [==============================] - 22s 707us/sample - loss: 1.3600 - val_loss: 1.3954\n",
      "Epoch 48/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3581\n",
      "Epoch 48: val_loss did not improve from 1.36816\n",
      "31305/31305 [==============================] - 22s 703us/sample - loss: 1.3581 - val_loss: 1.3938\n",
      "Epoch 49/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3571\n",
      "Epoch 49: val_loss did not improve from 1.36816\n",
      "31305/31305 [==============================] - 20s 651us/sample - loss: 1.3571 - val_loss: 1.3937\n",
      "Epoch 50/50\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3559\n",
      "Epoch 50: val_loss did not improve from 1.36816\n",
      "31305/31305 [==============================] - 22s 705us/sample - loss: 1.3559 - val_loss: 1.3911\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 21:46:34.320088: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_31_5/lstm_cell_216/recurrent_kernel/Assign' id:101138 op device:{requested: '', assigned: ''} def:{{{node lstm_31_5/lstm_cell_216/recurrent_kernel/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_31_5/lstm_cell_216/recurrent_kernel, lstm_31_5/lstm_cell_216/recurrent_kernel/Initializer/mul_1)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 21:46:39.404595: W tensorflow/c/c_api.cc:304] Operation '{name:'conv2d_3_5/bias/v/Assign' id:103406 op device:{requested: '', assigned: ''} def:{{{node conv2d_3_5/bias/v/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](conv2d_3_5/bias/v, conv2d_3_5/bias/v/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 21:46:44.360408: W tensorflow/c/c_api.cc:304] Operation '{name:'dropout_4_5/cond/Merge' id:102107 op device:{requested: '', assigned: ''} def:{{{node dropout_4_5/cond/Merge}} = Merge[N=2, T=DT_FLOAT, _has_manual_control_dependencies=true](dropout_4_5/cond/Identity, dropout_4_5/cond/dropout/SelectV2)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1485, 2208)\n",
      "(1514, 2208)\n",
      "(1644, 2208)\n",
      "(1764, 2208)\n",
      "(1836, 2208)\n",
      "(1699, 2208)\n",
      "(1369, 2208)\n",
      "(1766, 2208)\n",
      "(1631, 2208)\n",
      "(1692, 2208)\n",
      "(1550, 2208)\n",
      "(1644, 2208)\n",
      "(1800, 2208)\n",
      "(1872, 2208)\n",
      "(1716, 2208)\n",
      "(1788, 2208)\n",
      "(971, 2208)\n",
      "(1692, 2208)\n",
      "(1872, 2208)\n",
      "{1: 6.027674751480726, 2: 5.309021056666758, 4: 10.0, 5: 5.46722426118386, 6: 4.3242976794166665, 8: 8.62954164294796, 9: 6.685205465899126, 10: 7.497971293566844, 11: 6.656332241774235, 12: 9.422816566852315, 13: 6.416091519941443, 19: 8.39129453286052, 21: 9.188709024763133, 22: 1.0, 25: 7.779711447245096, 26: 6.444745420508052, 27: 4.304798895764984, 28: 5.5545872696902, 29: 1.4744323160640715}\n",
      "Train on 31305 samples, validate on 3488 samples\n",
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 21:51:00.878678: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_9/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31305/31305 [==============================] - ETA: 0s - loss: 8.9229\n",
      "Epoch 1: val_loss improved from inf to 1.38142, saving model to ./checkpoints/unknown_person_p17_22.h5\n",
      "31305/31305 [==============================] - 33s 1ms/sample - loss: 8.9229 - val_loss: 1.3814\n",
      "Epoch 2/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.9198\n",
      "Epoch 2: val_loss improved from 1.38142 to 1.38033, saving model to ./checkpoints/unknown_person_p17_22.h5\n",
      "31305/31305 [==============================] - 22s 700us/sample - loss: 8.9198 - val_loss: 1.3803\n",
      "Epoch 3/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.9666\n",
      "Epoch 3: val_loss did not improve from 1.38033\n",
      "31305/31305 [==============================] - 20s 633us/sample - loss: 8.9666 - val_loss: 1.3826\n",
      "Epoch 4/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.8751\n",
      "Epoch 4: val_loss improved from 1.38033 to 1.37960, saving model to ./checkpoints/unknown_person_p17_22.h5\n",
      "31305/31305 [==============================] - 20s 631us/sample - loss: 8.8751 - val_loss: 1.3796\n",
      "Epoch 5/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.8355\n",
      "Epoch 5: val_loss did not improve from 1.37960\n",
      "31305/31305 [==============================] - 20s 625us/sample - loss: 8.8355 - val_loss: 1.3871\n",
      "Epoch 6/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.7985\n",
      "Epoch 6: val_loss did not improve from 1.37960\n",
      "31305/31305 [==============================] - 20s 641us/sample - loss: 8.7985 - val_loss: 1.3878\n",
      "Epoch 7/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.7825\n",
      "Epoch 7: val_loss did not improve from 1.37960\n",
      "31305/31305 [==============================] - 21s 685us/sample - loss: 8.7825 - val_loss: 1.3854\n",
      "Epoch 8/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.7797\n",
      "Epoch 8: val_loss improved from 1.37960 to 1.37102, saving model to ./checkpoints/unknown_person_p17_22.h5\n",
      "31305/31305 [==============================] - 20s 629us/sample - loss: 8.7797 - val_loss: 1.3710\n",
      "Epoch 9/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.7353\n",
      "Epoch 9: val_loss did not improve from 1.37102\n",
      "31305/31305 [==============================] - 20s 626us/sample - loss: 8.7353 - val_loss: 1.3795\n",
      "Epoch 10/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.7517\n",
      "Epoch 10: val_loss did not improve from 1.37102\n",
      "31305/31305 [==============================] - 20s 639us/sample - loss: 8.7517 - val_loss: 1.3794\n",
      "Epoch 11/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.7195\n",
      "Epoch 11: val_loss did not improve from 1.37102\n",
      "31305/31305 [==============================] - 20s 631us/sample - loss: 8.7195 - val_loss: 1.3740\n",
      "Epoch 12/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.7534\n",
      "Epoch 12: val_loss did not improve from 1.37102\n",
      "31305/31305 [==============================] - 21s 682us/sample - loss: 8.7534 - val_loss: 1.3813\n",
      "Epoch 13/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.7357\n",
      "Epoch 13: val_loss did not improve from 1.37102\n",
      "31305/31305 [==============================] - 21s 678us/sample - loss: 8.7357 - val_loss: 1.3785\n",
      "Epoch 14/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.7129\n",
      "Epoch 14: val_loss did not improve from 1.37102\n",
      "31305/31305 [==============================] - 20s 629us/sample - loss: 8.7129 - val_loss: 1.3779\n",
      "Epoch 15/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.7123\n",
      "Epoch 15: val_loss did not improve from 1.37102\n",
      "31305/31305 [==============================] - 20s 642us/sample - loss: 8.7123 - val_loss: 1.3767\n",
      "Epoch 16/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.7075\n",
      "Epoch 16: val_loss did not improve from 1.37102\n",
      "31305/31305 [==============================] - 22s 690us/sample - loss: 8.7075 - val_loss: 1.3778\n",
      "Epoch 17/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.7018\n",
      "Epoch 17: val_loss did not improve from 1.37102\n",
      "31305/31305 [==============================] - 22s 711us/sample - loss: 8.7018 - val_loss: 1.3719\n",
      "Epoch 18/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.6925\n",
      "Epoch 18: val_loss did not improve from 1.37102\n",
      "31305/31305 [==============================] - 21s 658us/sample - loss: 8.6925 - val_loss: 1.3787\n",
      "Epoch 19/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.6759\n",
      "Epoch 19: val_loss did not improve from 1.37102\n",
      "31305/31305 [==============================] - 20s 632us/sample - loss: 8.6759 - val_loss: 1.3762\n",
      "Epoch 20/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 8.6826\n",
      "Epoch 20: val_loss did not improve from 1.37102\n",
      "31305/31305 [==============================] - 20s 626us/sample - loss: 8.6826 - val_loss: 1.3760\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 21:58:04.028691: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_13_6/lstm_cell_235/bias/Assign' id:117663 op device:{requested: '', assigned: ''} def:{{{node lstm_13_6/lstm_cell_235/bias/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_13_6/lstm_cell_235/bias, lstm_13_6/lstm_cell_235/bias/Initializer/concat)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 21:58:09.950588: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_30_6/lstm_cell_252/recurrent_kernel/m/Assign' id:122621 op device:{requested: '', assigned: ''} def:{{{node lstm_30_6/lstm_cell_252/recurrent_kernel/m/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_30_6/lstm_cell_252/recurrent_kernel/m, lstm_30_6/lstm_cell_252/recurrent_kernel/m/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Train on 31305 samples, validate on 3488 samples\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 21:58:18.711397: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_13/mul' id:122095 op device:{requested: '', assigned: ''} def:{{{node loss_13/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_13/mul/x, loss_13/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 21:58:50.072594: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_10/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3248"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 21:59:11.150775: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_13/mul' id:122095 op device:{requested: '', assigned: ''} def:{{{node loss_13/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_13/mul/x, loss_13/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Epoch 1: val_loss improved from inf to 1.36829, saving model to ./checkpoints/unknown_person_baseline_2_p17_22.h5\n",
      "31305/31305 [==============================] - 33s 1ms/sample - loss: 1.3248 - val_loss: 1.3683\n",
      "Epoch 2/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3208\n",
      "Epoch 2: val_loss improved from 1.36829 to 1.36728, saving model to ./checkpoints/unknown_person_baseline_2_p17_22.h5\n",
      "31305/31305 [==============================] - 20s 644us/sample - loss: 1.3208 - val_loss: 1.3673\n",
      "Epoch 3/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3468\n",
      "Epoch 3: val_loss did not improve from 1.36728\n",
      "31305/31305 [==============================] - 22s 690us/sample - loss: 1.3468 - val_loss: 1.3707\n",
      "Epoch 4/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3356\n",
      "Epoch 4: val_loss did not improve from 1.36728\n",
      "31305/31305 [==============================] - 22s 689us/sample - loss: 1.3356 - val_loss: 1.3704\n",
      "Epoch 5/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3301\n",
      "Epoch 5: val_loss did not improve from 1.36728\n",
      "31305/31305 [==============================] - 19s 610us/sample - loss: 1.3301 - val_loss: 1.3725\n",
      "Epoch 6/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3314\n",
      "Epoch 6: val_loss did not improve from 1.36728\n",
      "31305/31305 [==============================] - 22s 690us/sample - loss: 1.3314 - val_loss: 1.3837\n",
      "Epoch 7/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3291\n",
      "Epoch 7: val_loss did not improve from 1.36728\n",
      "31305/31305 [==============================] - 20s 648us/sample - loss: 1.3291 - val_loss: 1.3696\n",
      "Epoch 8/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3260\n",
      "Epoch 8: val_loss did not improve from 1.36728\n",
      "31305/31305 [==============================] - 19s 614us/sample - loss: 1.3260 - val_loss: 1.3721\n",
      "Epoch 9/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3224\n",
      "Epoch 9: val_loss did not improve from 1.36728\n",
      "31305/31305 [==============================] - 19s 617us/sample - loss: 1.3224 - val_loss: 1.3739\n",
      "Epoch 10/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3226\n",
      "Epoch 10: val_loss did not improve from 1.36728\n",
      "31305/31305 [==============================] - 19s 617us/sample - loss: 1.3226 - val_loss: 1.3715\n",
      "Epoch 11/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3214\n",
      "Epoch 11: val_loss did not improve from 1.36728\n",
      "31305/31305 [==============================] - 21s 676us/sample - loss: 1.3214 - val_loss: 1.3679\n",
      "Epoch 12/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3207\n",
      "Epoch 12: val_loss did not improve from 1.36728\n",
      "31305/31305 [==============================] - 22s 688us/sample - loss: 1.3207 - val_loss: 1.3719\n",
      "Epoch 13/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3189\n",
      "Epoch 13: val_loss did not improve from 1.36728\n",
      "31305/31305 [==============================] - 22s 703us/sample - loss: 1.3189 - val_loss: 1.3686\n",
      "Epoch 14/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3191\n",
      "Epoch 14: val_loss did not improve from 1.36728\n",
      "31305/31305 [==============================] - 22s 706us/sample - loss: 1.3191 - val_loss: 1.3696\n",
      "Epoch 15/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3211\n",
      "Epoch 15: val_loss did not improve from 1.36728\n",
      "31305/31305 [==============================] - 21s 663us/sample - loss: 1.3211 - val_loss: 1.3697\n",
      "Epoch 16/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3217\n",
      "Epoch 16: val_loss did not improve from 1.36728\n",
      "31305/31305 [==============================] - 19s 619us/sample - loss: 1.3217 - val_loss: 1.3683\n",
      "Epoch 17/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3205\n",
      "Epoch 17: val_loss did not improve from 1.36728\n",
      "31305/31305 [==============================] - 22s 693us/sample - loss: 1.3205 - val_loss: 1.3815\n",
      "Epoch 18/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3168\n",
      "Epoch 18: val_loss improved from 1.36728 to 1.36375, saving model to ./checkpoints/unknown_person_baseline_2_p17_22.h5\n",
      "31305/31305 [==============================] - 21s 674us/sample - loss: 1.3168 - val_loss: 1.3638\n",
      "Epoch 19/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3171\n",
      "Epoch 19: val_loss did not improve from 1.36375\n",
      "31305/31305 [==============================] - 20s 650us/sample - loss: 1.3171 - val_loss: 1.3674\n",
      "Epoch 20/20\n",
      "31305/31305 [==============================] - ETA: 0s - loss: 1.3169\n",
      "Epoch 20: val_loss did not improve from 1.36375\n",
      "31305/31305 [==============================] - 22s 696us/sample - loss: 1.3169 - val_loss: 1.3677\n"
     ]
    }
   ],
   "source": [
    "person_nums = [1,2,4,5,6,8,9,10,11,12,13,17,19,21,22,25,26,27,28,29]\n",
    "## Build the baseline model for emotion recognition with dropout layers\n",
    "from tensorflow.keras.models import Model\n",
    "from tensorflow.keras.layers import Input, Dense, concatenate, SimpleRNN, LSTM, Conv2D, Flatten, MaxPooling2D, GRU, AveragePooling2D, Dropout, Lambda\n",
    "from tensorflow.keras.callbacks import ModelCheckpoint\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "def individual_model(features_list):\n",
    "    input_layers = []\n",
    "    hidden_layers = []\n",
    "    combined_feature = np.empty((len(features_list[0]),0))\n",
    "    for i, feature in enumerate(features_list):\n",
    "        \n",
    "        if len(feature.shape) == 3:\n",
    "            input_i = Input(shape=(feature.shape[1], feature.shape[2]))\n",
    "            input_layers.append(input_i)\n",
    "\n",
    "            hidden_i = input_i[:,:,:,None]\n",
    "            hidden_i = Conv2D(32, 3, activation='relu')(hidden_i)\n",
    "            hidden_i = AveragePooling2D((3, 3))(hidden_i)\n",
    "            hidden_i = Dropout(0.5)(hidden_i)\n",
    "            hidden_i = Conv2D(16, 3, activation='relu')(hidden_i)\n",
    "            hidden_i = AveragePooling2D((3, 3))(hidden_i)\n",
    "            hidden_i = Dropout(0.5)(hidden_i)\n",
    "            hidden_i = Conv2D(8, 1, activation='relu')(hidden_i)\n",
    "            hidden_i = AveragePooling2D((2, 2))(hidden_i)\n",
    "            hidden_i = Dropout(0.5)(hidden_i)\n",
    "            hidden_i = Conv2D(4, 1, activation='relu')(hidden_i)\n",
    "            hidden_i = AveragePooling2D((2, 2))(hidden_i)\n",
    "            hidden_i = Dropout(0.5)(hidden_i)\n",
    "            hidden_i = Flatten()(hidden_i)\n",
    "\n",
    "            hidden_layers.append(hidden_i)\n",
    "        elif feature.shape[1] <20:\n",
    "            combined_feature = np.hstack((combined_feature, feature))\n",
    "            \n",
    "        else:  # For series features\n",
    "            input_i = Input(shape=(feature.shape[1],))\n",
    "            input_layers.append(input_i)\n",
    "            hidden_i = Lambda(lambda x: x[:, :, None])(input_i)  # Add a new dimension\n",
    "            hidden_i = LSTM(4)(hidden_i)\n",
    "            hidden_layers.append(hidden_i)\n",
    "    input_i = Input(shape=(combined_feature.shape[1],))\n",
    "    input_layers.append(input_i)\n",
    "    dense_num = np.max((1, int(combined_feature.shape[1]/2)))\n",
    "    hidden_i = Dense(dense_num, activation='relu')(input_i)\n",
    "    hidden_layers.append(hidden_i)\n",
    "    print(combined_feature.shape)\n",
    "    concat_layer = concatenate(hidden_layers)\n",
    "    h = Dropout(0.2)(concat_layer)\n",
    "    h = Dense(64, activation='relu')(h)\n",
    "    h = Dense(32, activation='relu')(h)\n",
    "    output_layer = Dense(2)(h)\n",
    "    model = Model(inputs=input_layers, outputs=output_layer)\n",
    "    model.compile(loss='mean_absolute_error', optimizer='adam')\n",
    "    return model\n",
    "\n",
    "import tensorflow as tf\n",
    "from tensorflow.keras.callbacks import Callback\n",
    "import numpy as np\n",
    "\n",
    "class PruningCallback(tf.keras.callbacks.Callback):\n",
    "    def __init__(self, initial_pruning_factor=0.1, final_pruning_factor=0.5, start_epoch=0, end_epoch=None, frequency=1):\n",
    "        super(PruningCallback, self).__init__()\n",
    "        self.initial_pruning_factor = initial_pruning_factor\n",
    "        self.final_pruning_factor = final_pruning_factor\n",
    "        self.start_epoch = start_epoch\n",
    "        self.end_epoch = end_epoch if end_epoch is not None else np.inf\n",
    "        self.frequency = frequency\n",
    "        self.pruned_weights = {}\n",
    "        self.layer_importance = {}\n",
    "\n",
    "    def get_pruning_factor(self, epoch):\n",
    "        if epoch < self.start_epoch:\n",
    "            return 0\n",
    "        if epoch > self.end_epoch:\n",
    "            return self.final_pruning_factor\n",
    "        return self.initial_pruning_factor + (self.final_pruning_factor - self.initial_pruning_factor) * (epoch - self.start_epoch) / (self.end_epoch - self.start_epoch)\n",
    "\n",
    "    def on_train_begin(self, logs=None):\n",
    "        total_weight_magnitude = 0\n",
    "        for layer in self.model.layers:\n",
    "            if hasattr(layer, 'get_weights'):\n",
    "                weights = layer.get_weights()\n",
    "                layer_norm = sum(np.linalg.norm(w) for w in weights)\n",
    "                total_weight_magnitude += layer_norm\n",
    "                self.layer_importance[layer.name] = layer_norm\n",
    "    \n",
    "        # Normalize the layer importance values so they sum up to 1\n",
    "        for layer_name in self.layer_importance:\n",
    "            self.layer_importance[layer_name] /= total_weight_magnitude\n",
    "    # def on_train_begin(self, logs=None):\n",
    "    #     total_weight_magnitude = sum([np.linalg.norm(layer.get_weights()) for layer in self.model.layers if hasattr(layer, 'get_weights')])\n",
    "    #     for layer in self.model.layers:\n",
    "    #         if hasattr(layer, 'get_weights'):\n",
    "    #             self.layer_importance[layer.name] = np.linalg.norm(layer.get_weights()) / total_weight_magnitude\n",
    "\n",
    "    def get_layer_pruning_factor(self, layer_name, global_pruning_factor):\n",
    "        if layer_name in self.layer_importance:\n",
    "            importance = self.layer_importance[layer_name]\n",
    "            adjusted_pruning_factor = global_pruning_factor * (1 - importance)\n",
    "            return min(adjusted_pruning_factor, 1)  # Ensure the pruning factor is not greater than 1\n",
    "        return global_pruning_factor\n",
    "    def prune_weights(self, layer, global_pruning_factor):\n",
    "        \n",
    "        weights = layer.get_weights()\n",
    "        layer_name = layer.name\n",
    "        pruning_factor = self.get_layer_pruning_factor(layer_name, global_pruning_factor)\n",
    "\n",
    "        if layer_name not in self.pruned_weights:\n",
    "            self.pruned_weights[layer_name] = [np.zeros_like(w, dtype=bool) for w in weights]\n",
    "        \n",
    "        for i in range(len(weights)):\n",
    "            weight = weights[i]\n",
    "            # print(weight.shape)\n",
    "            # print(weight.size)\n",
    "            if weight.ndim > 1:  # Only prune dense or convolutional layers\n",
    "                unpruned_weights = np.logical_not(self.pruned_weights[layer_name][i])\n",
    "                num_unpruned = np.sum(unpruned_weights)\n",
    "                num_pruning = min(num_unpruned, int(weight.size * pruning_factor) - np.sum(self.pruned_weights[layer_name][i]))\n",
    "                num_pruning = int(weight.size * pruning_factor) - np.sum(self.pruned_weights[layer_name][i])\n",
    "                if num_pruning > 0:\n",
    "                    unpruned_flat_indices = np.flatnonzero(unpruned_weights)\n",
    "                    abs_unpruned_weights = np.abs(weight[unpruned_weights])\n",
    "                    pruning_flat_indices = np.argpartition(abs_unpruned_weights, num_pruning)[:num_pruning]\n",
    "                    \n",
    "                    indices = np.unravel_index(pruning_flat_indices, weight.shape)\n",
    "                    self.pruned_weights[layer_name][i][indices] = True\n",
    "\n",
    "                weights[i] = weights[i]*(~self.pruned_weights[layer_name][i])\n",
    "                \n",
    "        layer.set_weights(weights)\n",
    "\n",
    "    def on_epoch_end(self, epoch, logs=None):\n",
    "        if (epoch - self.start_epoch) % self.frequency != 0:\n",
    "            return\n",
    "\n",
    "        pruning_factor = self.get_pruning_factor(epoch)\n",
    "        for layer in self.model.layers:\n",
    "            if isinstance(layer, (tf.keras.layers.Dense, tf.keras.layers.Conv2D)):\n",
    "                self.prune_weights(layer, pruning_factor)\n",
    "from feature_emotion import feature_extract, split_data, label_unique_tuples, split_data_unknown\n",
    "gts, sensor_nums, walk_nums, trace_nums, people_nums, spe_centr, delta_spe_centr, spe_crest, delta_spe_crest, spe_decrease, delta_spe_decrease, spe_entropy, delta_spe_entropy, spe_flatness, delta_spe_flatness, spe_flux, delta_spe_flux, spe_kurtosis, delta_spe_kurtosis, spe_skewness, delta_spe_skewness, spe_rfp, delta_spe_rfp, spe_slope, delta_spe_slope, spe_spread, delta_spe_spread, wlk_fre, wlk_fres_trace, cwt_figs_all, cwt_sum_all_0, cwt_sum_all_1, cwt_sum_all_2, cwt_sum_all_3, high_fre_compos, pitchs, low_fre_compos, auto_corrs, real_hils, imag_hils, dur_time_1_alls, dur_time_2_alls, jitters, shimmers, jitter_rap, hrs, feature, slope, zcrs, fft_features, energy_alls, log_energy_alls, smoothe_energy_alls, legendres, double_support_time, pdps_new, lpcs, ceps_features = feature_extract(person_nums)\n",
    "\n",
    "walk_nums_all = np.squeeze(walk_nums)\n",
    "trace_nums_all = np.squeeze(trace_nums)\n",
    "people_nums_all = np.squeeze(people_nums)\n",
    "\n",
    "## 0: train, 1: validation 2: test\n",
    "\n",
    "test_person_id = [17]\n",
    "flag_tr_val_te = split_data_unknown(test_person_id, walk_nums_all, trace_nums_all, people_nums_all, person_nums, rand_seed = 42)\n",
    "## Data Normalization before training ans testing\n",
    "import tensorflow as tf\n",
    "tf.compat.v1.disable_v2_behavior()\n",
    "from tensorflow.keras.models import Model\n",
    "from tensorflow.keras.layers import Input, Dense, concatenate, LSTM\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "scalers = []\n",
    "X_train_normalized = []\n",
    "X_val_normalized = []\n",
    "X_test_normalized = []\n",
    "train_idx = np.where(flag_tr_val_te ==0)[0]\n",
    "np.random.shuffle(train_idx)\n",
    "val_idx = np.where(flag_tr_val_te ==1)[0]\n",
    "test_idx = np.where(flag_tr_val_te ==2)[0]\n",
    "\n",
    "for i, feature in enumerate([spe_centr, delta_spe_centr, spe_crest, delta_spe_crest, spe_decrease, delta_spe_decrease, spe_entropy, delta_spe_entropy, spe_flatness, delta_spe_flatness, spe_flux, delta_spe_flux, spe_kurtosis, delta_spe_kurtosis, spe_skewness, delta_spe_skewness, spe_rfp, delta_spe_rfp, spe_slope, delta_spe_slope, spe_spread, delta_spe_spread, wlk_fre, wlk_fres_trace, cwt_figs_all, cwt_sum_all_0, cwt_sum_all_1, cwt_sum_all_2, cwt_sum_all_3, high_fre_compos, pitchs, low_fre_compos, auto_corrs, real_hils, imag_hils, dur_time_1_alls, dur_time_2_alls, jitters, shimmers, jitter_rap, hrs, feature, slope, zcrs, fft_features, energy_alls, log_energy_alls, smoothe_energy_alls, legendres, double_support_time, pdps_new, lpcs, ceps_features]):\n",
    "    scaler = StandardScaler()\n",
    "    if len(feature.shape)==2:\n",
    "        X_train_i = feature[train_idx,:]\n",
    "        X_val_i = feature[val_idx,:]\n",
    "        X_test_i = feature[test_idx,:]\n",
    "        X_train_normalized_i = scaler.fit_transform(X_train_i)\n",
    "        X_val_normalized_i = scaler.transform(X_val_i)\n",
    "        X_test_normalized_i = scaler.transform(X_test_i)\n",
    "        scalers.append(scaler)\n",
    "    else:\n",
    "        X_train_i = feature[train_idx,:,:]\n",
    "        X_val_i = feature[val_idx,:,:]\n",
    "        X_test_i = feature[test_idx,:,:]\n",
    "        X_train_normalized_i = scaler.fit_transform(X_train_i.reshape(X_train_i.shape[0], -1)).reshape(X_train_i.shape)\n",
    "        X_val_normalized_i = scaler.transform(X_val_i.reshape(X_val_i.shape[0], -1)).reshape(X_val_i.shape)\n",
    "        X_test_normalized_i = scaler.transform(X_test_i.reshape(X_test_i.shape[0], -1)).reshape(X_test_i.shape)\n",
    "        scalers.append(scaler)\n",
    "    X_train_normalized.append(X_train_normalized_i)\n",
    "    X_val_normalized.append(X_val_normalized_i)\n",
    "    X_test_normalized.append(X_test_normalized_i)\n",
    "y_train = gts[train_idx,:]\n",
    "y_val = gts[val_idx,:]\n",
    "y_test = gts[test_idx,:]\n",
    "X_train_normalized_new = []\n",
    "combined_feature = np.empty((len(X_train_normalized[0]),0))\n",
    "for feature in X_train_normalized:\n",
    "    if len(feature.shape) == 3:\n",
    "        X_train_normalized_new.append(feature)\n",
    "    elif feature.shape[1] <20:\n",
    "        combined_feature = np.hstack((combined_feature, feature))\n",
    "    else:\n",
    "        X_train_normalized_new.append(feature)\n",
    "X_train_normalized_new.append(combined_feature)\n",
    "\n",
    "X_val_normalized_new = []\n",
    "combined_feature = np.empty((len(X_val_normalized[0]),0))\n",
    "for feature in X_val_normalized:\n",
    "    if len(feature.shape) == 3:\n",
    "        X_val_normalized_new.append(feature)\n",
    "    elif feature.shape[1] <20:\n",
    "        combined_feature = np.hstack((combined_feature, feature))\n",
    "    else:\n",
    "        X_val_normalized_new.append(feature)\n",
    "X_val_normalized_new.append(combined_feature)\n",
    "\n",
    "X_test_normalized_new = []\n",
    "combined_feature = np.empty((len(X_test_normalized[0]),0))\n",
    "for feature in X_test_normalized:\n",
    "    if len(feature.shape) == 3:\n",
    "        X_test_normalized_new.append(feature)\n",
    "    elif feature.shape[1] <20:\n",
    "        combined_feature = np.hstack((combined_feature, feature))\n",
    "    else:\n",
    "        X_test_normalized_new.append(feature)\n",
    "X_test_normalized_new.append(combined_feature)\n",
    "\n",
    "model = individual_model(X_train_normalized)\n",
    "model.compile(loss='mean_absolute_error', optimizer='adam')\n",
    "num_epochs = 50\n",
    "\n",
    "\n",
    "# Gradually prune weights in dense and convolutional layers from 10% to 50% over the course of training, starting from epoch 0.\n",
    "\n",
    "\n",
    "rates = [0.4, 0.5, 0.6]\n",
    "idx = 20\n",
    "for r in rates:\n",
    "    model_name = './checkpoints/unknown_person_baseline_p'+str(test_person_id[0])+'_'+str(idx)+'.h5'\n",
    "    checkpoint = ModelCheckpoint(model_name, monitor='val_loss', mode='min', save_best_only=True, verbose=1)\n",
    "\n",
    "    pruning_callback = PruningCallback(initial_pruning_factor=0.01, final_pruning_factor=r, start_epoch=5, end_epoch=20, frequency=1)\n",
    "    model.fit(X_train_normalized_new, y_train, epochs=num_epochs, batch_size=1024, validation_data=(X_val_normalized_new, y_val), callbacks=[pruning_callback, checkpoint])\n",
    "    import tensorflow as tf\n",
    "    model = tf.keras.models.load_model(model_name)\n",
    "    from tensorflow.keras.models import Model\n",
    "    layers = model.layers\n",
    "    second_last_layer_output = layers[-4].output\n",
    "    feature_extractor_model = Model(inputs=model.input, outputs=second_last_layer_output)\n",
    "    train_features = feature_extractor_model.predict(X_train_normalized_new)\n",
    "    test_features = feature_extractor_model.predict(X_test_normalized_new)\n",
    "    \n",
    "    p_train = people_nums[train_idx,:]\n",
    "    p_val = people_nums[val_idx,:]\n",
    "    p_test = people_nums[test_idx,:]\n",
    "    ## Calculate the distance between test person and training person\n",
    "    def euclidean_distance(a, b):\n",
    "        return np.sqrt(np.sum((a - b) ** 2, axis=1))\n",
    "    \n",
    "    distance_dict = {}\n",
    "    for ii in range(len(person_nums)):\n",
    "        if person_nums[ii] == test_person_id[0]:\n",
    "            continue\n",
    "        else:\n",
    "            \n",
    "            ind = np.where(p_train ==person_nums[ii])[0]\n",
    "            tmp_train_features = train_features[ind, :]\n",
    "            distances = np.array([euclidean_distance(train_sample, test_features) for train_sample in tmp_train_features])\n",
    "            print(distances.shape)\n",
    "            average_distances = np.mean(distances, axis=1)\n",
    "    \n",
    "            # Step 4: Find the overall average distance\n",
    "            overall_average_distance = np.mean(average_distances)\n",
    "            distance_dict[person_nums[ii]] = overall_average_distance\n",
    "    \n",
    "    \n",
    "    def normalize_to_weights(distance_dict):\n",
    "        distances = np.array(list(distance_dict.values()))\n",
    "        # Handle the case where a distance is zero to avoid division by zero\n",
    "        distances = np.clip(distances, a_min=1e-10, a_max=None)\n",
    "        weights = 1 / distances\n",
    "        normalized_weights = weights\n",
    "        # normalized_weights = weights / sum(weights)\n",
    "        # print(sum(weights))\n",
    "        # print(sum(normalized_weights))\n",
    "        # Assign the normalized weights back to the dictionary\n",
    "        normalized_weight_dict = dict(zip(distance_dict.keys(), normalized_weights))\n",
    "        return normalized_weight_dict\n",
    "    def scale_dict_values(my_dict):\n",
    "        scaled_dict = my_dict.copy()\n",
    "        min_val = min(scaled_dict.values())\n",
    "        max_val = max(scaled_dict.values())\n",
    "        \n",
    "        for key in scaled_dict:\n",
    "            scaled_dict[key] = 1 + 9 * (scaled_dict[key] - min_val) / (max_val - min_val)\n",
    "        \n",
    "        return scaled_dict\n",
    "    weights_dict = normalize_to_weights(distance_dict)\n",
    "    weights_dict = scale_dict_values(weights_dict)\n",
    "    print(weights_dict)\n",
    "    \n",
    "    w_train = np.zeros_like(p_train)\n",
    "    for i in range(len(w_train)):\n",
    "        w_train[i] = weights_dict[int(p_train[i])]\n",
    "    \n",
    "    w_train = np.squeeze(w_train)\n",
    "    \n",
    "    import sys\n",
    "    sys.path.append('..')\n",
    "    # for layer in model.layers[-4:]:\n",
    "    #     layer.trainable = False\n",
    "    # model = individual_model.individual_model(X_train_normalized)\n",
    "    model_name = './checkpoints/unknown_person_p'+str(test_person_id[0])+'_'+str(idx)+'.h5'\n",
    "    checkpoint = ModelCheckpoint(model_name, monitor='val_loss', mode='min', save_best_only=True, verbose=1)\n",
    "    pruning_callback = PruningCallback(initial_pruning_factor=r, final_pruning_factor=r, start_epoch=1, end_epoch=20, frequency=1)\n",
    "    \n",
    "    history = model.fit(x=X_train_normalized_new, y=y_train,sample_weight= w_train,epochs=20, batch_size=1024, validation_data=(X_val_normalized_new, y_val), callbacks=[pruning_callback, checkpoint])\n",
    "    import sys\n",
    "    sys.path.append('..')\n",
    "    # for layer in model.layers[-4:]:\n",
    "    #     layer.trainable = False\n",
    "    # model = individual_model.individual_model(X_train_normalized)\n",
    "    model_name = './checkpoints/unknown_person_baseline_p'+str(test_person_id[0])+'_'+str(idx)+'.h5'\n",
    "    model = tf.keras.models.load_model(model_name)\n",
    "    model_name = './checkpoints/unknown_person_baseline_2_p'+str(test_person_id[0])+'_'+str(idx)+'.h5'\n",
    "    idx += 1\n",
    "    checkpoint = ModelCheckpoint(model_name, monitor='val_loss', mode='min', save_best_only=True, verbose=1)\n",
    "    pruning_callback = PruningCallback(initial_pruning_factor=r, final_pruning_factor=r, start_epoch=1, end_epoch=20, frequency=1)\n",
    "    \n",
    "    history = model.fit(x=X_train_normalized_new, y=y_train,epochs=20, batch_size=1024, validation_data=(X_val_normalized_new, y_val), callbacks=[pruning_callback, checkpoint])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f71b59e4-6581-4ea8-994d-a02a6654ab6b",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
